diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java b/src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java
index 0d05ac1b5..064d46de4 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java
@@ -184,7 +184,8 @@ public final class BlockCommentPosition {
     public static boolean isOnConstructor(DetailAST blockComment) {
         return isOnPlainToken(blockComment, TokenTypes.CTOR_DEF, TokenTypes.IDENT)
                 || isOnTokenWithModifiers(blockComment, TokenTypes.CTOR_DEF)
-                || isOnTokenWithAnnotation(blockComment, TokenTypes.CTOR_DEF);
+                || isOnTokenWithAnnotation(blockComment, TokenTypes.CTOR_DEF)
+                || isOnPlainClassMember(blockComment, TokenTypes.CTOR_DEF);
     }
 
     /**
diff --git a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/invalidjavadocposition/InputInvalidJavadocPosition.java b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/invalidjavadocposition/InputInvalidJavadocPosition.java
index 6b801f42e..0d8c88bf8 100644
--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/invalidjavadocposition/InputInvalidJavadocPosition.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/invalidjavadocposition/InputInvalidJavadocPosition.java
@@ -4,24 +4,24 @@ InvalidJavadocPosition
 
 */
 
-package /** violation */ com.puppycrawl.tools // violation
+package /** invalid */ com.puppycrawl.tools // violation 'Javadoc .* placed in the wrong location.'
         .checkstyle.checks.javadoc.invalidjavadocposition;
 
-/** violation */ // violation
+/** invalid */ // violation 'Javadoc .* placed in the wrong location.'
 import java.lang.String;
 
-/** violation */ // violation
+/** invalid */ // violation 'Javadoc .* placed in the wrong location.'
 /** valid */
 class InputInvalidJavadocPosition {
-    /** violation */ // violation
+    /** invalid */ // violation 'Javadoc .* placed in the wrong location.'
 }
 /** valid */
 /* ignore */
 class InputInvalidJavadocPosition2 {
-    /** violation */ // violation
+    /** invalid */ // violation 'Javadoc .* placed in the wrong location.'
     static { /* ignore */ }
 
-    /** violation */ // violation
+    /** invalid */ // violation 'Javadoc .* placed in the wrong location.'
     /** valid */
     int field1;
     /** valid */
@@ -31,9 +31,9 @@ class InputInvalidJavadocPosition2 {
     /** valid */
     @Deprecated int field4;
 
-    int /** violation */ field20; // violation
-    int field21 /** violation */; // violation
-    @Deprecated /** violation */ int field22; // violation
+    int /** invalid */ field20; // violation 'Javadoc .* placed in the wrong location.'
+    int field21 /** invalid */; // violation 'Javadoc .* placed in the wrong location.'
+    @Deprecated /** invalid */ int field22; // violation 'Javadoc .* placed in the wrong location.'
 
     void method1() {}
     /** valid */
@@ -43,26 +43,52 @@ class InputInvalidJavadocPosition2 {
     /** valid */
     String[] method4() { return null; }
 
-    void /** violation */ method20() {} // violation
-    void method21 /** violation */ () {} // violation
-    void method22(/** violation */) {} // violation
-    void method23() /** violation */ {} // violation
-    void method24() { /** violation */ } // violation
-    void method25() { /** violation */ int variable; } // violation
+    void /** invalid */ method20() {} // violation 'Javadoc .* placed in the wrong location.'
+    void method21 /** invalid */ () {} // violation 'Javadoc .* placed in the wrong location.'
+    void method22(/** invalid */) {} // violation 'Javadoc .* placed in the wrong location.'
+    void method23() /** invalid */ {} // violation 'Javadoc .* placed in the wrong location.'
+    void method24() { /** invalid */ } // violation 'Javadoc .* placed in the wrong location.'
+    void method25() { /** invalid */ int variable; } // violation '.* placed in the wrong location.'
 }
 @Deprecated
-/** violation */ // violation
+/** invalid */ // violation 'Javadoc .* placed in the wrong location.'
 class InputInvalidJavadocPosition3 {}
 /** valid */
 @Deprecated
 class InputInvalidJavadocPosition4 {}
-class /** violation */ InputInvalidJavadocPosition5 {} // violation
-class InputInvalidJavadocPosition6 /** violation */ {} // violation
+class /** invalid */ InputInvalidJavadocPosition5 {} // violation '.* placed in the wrong location.'
+class InputInvalidJavadocPosition6 /** invalid */ {} // violation '.* placed in the wrong location.'
 class InputInvalidJavadocPosition7 {
     void method() {
-        /** violation */ int variable1; // violation
-        /** violation */ final int variable2; // violation
-        /** violation */ @Deprecated int variable3; // violation
+        /** invalid */ int variable1; // violation 'Javadoc .* placed in the wrong location.'
+        /** invalid */ final int variable2; // violation 'Javadoc .* placed in the wrong location.'
+        /** invalid */ @Deprecated int variable3; // violation '.* placed in the wrong location.'
     }
 }
-/** violation */ // violation
+class GenericConstructor {
+    /** valid */
+    <E extends String> GenericConstructor(E a) {}
+
+    // violation below 'Javadoc .* placed in the wrong location.'
+    </** invalid */E extends String>  GenericConstructor(E a, E b) {}
+
+    // violation below 'Javadoc .* placed in the wrong location.'
+    <E extends String> /** invalid */ GenericConstructor(E a, E b, E c) {}
+
+    // violation below 'Javadoc .* placed in the wrong location.'
+    <E extends String> GenericConstructor(/** invalid */E a, E b, E c, E d) {}
+
+    // violation below 'Javadoc .* placed in the wrong location.'
+    <E extends String> GenericConstructor(E a, E b, E c, E d, E e) {/** invalid */}
+
+    /** valid */
+    private <E extends String> GenericConstructor() {}
+
+    private class InnerClass {
+        /** valid */
+        <E extends String> InnerClass() {}
+
+        /** valid */ <E extends String> InnerClass(E a) {}
+    }
+}
+/** invalid */ // violation 'Javadoc .* placed in the wrong location.'
diff --git a/src/test/resources/com/puppycrawl/tools/checkstyle/utils/blockcommentposition/InputBlockCommentPositionOnConstructor.java b/src/test/resources/com/puppycrawl/tools/checkstyle/utils/blockcommentposition/InputBlockCommentPositionOnConstructor.java
index 85b0b376b..b138e8d18 100644
--- a/src/test/resources/com/puppycrawl/tools/checkstyle/utils/blockcommentposition/InputBlockCommentPositionOnConstructor.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/utils/blockcommentposition/InputBlockCommentPositionOnConstructor.java
@@ -24,4 +24,18 @@ public class InputBlockCommentPositionOnConstructor {
 
     }
 
+    /**
+     * I'm a javadoc
+     */
+    <T> InputBlockCommentPositionOnConstructor(boolean b){
+
+    }
+
+    /**
+     * I'm a javadoc
+     */
+    private <T> InputBlockCommentPositionOnConstructor(T t){
+
+    }
+
 }
