diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java
index dac05e893..d089a1698 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/OperatorWrapCheck.java
@@ -423,12 +423,7 @@ public class OperatorWrapCheck
             result = adjustParens(rightNode, DetailAST::getPreviousSibling);
         }
 
-        // The ARRAY_INIT AST is confusing. It should be
-        // ARRAY_INIT
-        // |--LCURLY (first child)
-        // `--RCURLY (last child)
-        // but there is no LCURLY for ARRAY_INIT
-        if (result.getType() != TokenTypes.ARRAY_INIT) {
+        if (!TokenUtil.isOfType(result, TokenTypes.ARRAY_INIT, TokenTypes.ANNOTATION_ARRAY_INIT)) {
             while (result.getFirstChild() != null) {
                 result = result.getFirstChild();
             }
diff --git a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapArrayAssign.java b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapArrayAssign.java
new file mode 100644
index 000000000..df6fb1bf3
--- /dev/null
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/operatorwrap/InputOperatorWrapArrayAssign.java
@@ -0,0 +1,77 @@
+/*
+OperatorWrap
+option = (default)nl
+tokens = ASSIGN
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.operatorwrap;
+
+import java.lang.annotation.Repeatable;
+import java.util.Arrays;
+
+@Annotation1(example = { // ok
+    "foo",
+    "bar"
+})
+@Annotation2(example = // violation
+{
+    "foo",
+    "bar"
+})
+@Nested(
+    nestedAnn = { // ok
+        @Annotation1(example = { // ok
+            "foo",
+            "bar"
+        })
+    }
+)
+public class InputOperatorWrapArrayAssign {
+    public String[] array = { // ok
+        "foo",
+        "bar"
+    };
+    public String[] array2 = // violation
+    {
+        "foo",
+        "bar"
+    };
+}
+
+@Annotation1(
+    value = "1",
+    example = { // ok
+        "foo",
+        "bar"
+    },
+    other = "2"
+)
+class Second {}
+
+@Annotation1(
+    example = { // ok
+        "foo",
+        "bar"
+    },
+    arr = { // ok
+        "1"
+    }
+)
+class Third {}
+
+@interface Annotation1 {
+    String[] example();
+    String value() default "";
+    String other() default "";
+    String[] arr() default "";
+}
+
+@interface Annotation2 {
+    String[] example();
+}
+
+@interface Nested {
+    Annotation1[] nestedAnn();
+}
