diff --git a/gson/src/main/java/com/google/gson/Gson.java b/gson/src/main/java/com/google/gson/Gson.java
index ba7bf4b4..c6f8508e 100644
--- a/gson/src/main/java/com/google/gson/Gson.java
+++ b/gson/src/main/java/com/google/gson/Gson.java
@@ -102,7 +102,7 @@ import java.util.concurrent.atomic.AtomicLongArray;
  * List&lt;MyType&gt; target2 = gson.fromJson(json, listType);
  * </pre>
  *
- * <p>See the <a href="https://github.com/google/gson/blob/master/UserGuide.md">Gson User Guide</a>
+ * <p>See the <a href="https://github.com/google/gson/blob/main/UserGuide.md">Gson User Guide</a>
  * for a more complete set of examples.</p>
  *
  * <h2 id="default-lenient">Lenient JSON handling</h2>
diff --git a/gson/src/main/java/com/google/gson/internal/TroubleshootingGuide.java b/gson/src/main/java/com/google/gson/internal/TroubleshootingGuide.java
index 69c1b06d..d0fbc1a4 100644
--- a/gson/src/main/java/com/google/gson/internal/TroubleshootingGuide.java
+++ b/gson/src/main/java/com/google/gson/internal/TroubleshootingGuide.java
@@ -7,6 +7,6 @@ public class TroubleshootingGuide {
    * Creates a URL referring to the specified troubleshooting section.
    */
   public static String createUrl(String id) {
-    return "https://github.com/google/gson/blob/master/Troubleshooting.md#" + id;
+    return "https://github.com/google/gson/blob/main/Troubleshooting.md#" + id;
   }
 }
diff --git a/shrinker-test/src/test/java/com/google/gson/it/ShrinkingIT.java b/shrinker-test/src/test/java/com/google/gson/it/ShrinkingIT.java
index ddf6f34f..9c32f1aa 100644
--- a/shrinker-test/src/test/java/com/google/gson/it/ShrinkingIT.java
+++ b/shrinker-test/src/test/java/com/google/gson/it/ShrinkingIT.java
@@ -194,7 +194,7 @@ public class ShrinkingIT {
         assertThat(e).hasCauseThat().hasMessageThat().isEqualTo(
             "Abstract classes can't be instantiated! Adjust the R8 configuration or register an InstanceCreator"
             + " or a TypeAdapter for this type. Class name: com.example.DefaultConstructorMain$TestClass"
-            + "\nSee https://github.com/google/gson/blob/master/Troubleshooting.md#r8-abstract-class"
+            + "\nSee https://github.com/google/gson/blob/main/Troubleshooting.md#r8-abstract-class"
         );
       }
     });
