diff --git a/naming/src/test/java/com/alibaba/nacos/naming/core/InstanceTest.java b/naming/src/test/java/com/alibaba/nacos/naming/core/InstanceTest.java
index 31cdad7..a2e04d6 100644
--- a/naming/src/test/java/com/alibaba/nacos/naming/core/InstanceTest.java
+++ b/naming/src/test/java/com/alibaba/nacos/naming/core/InstanceTest.java
@@ -16,6 +16,7 @@
 
 package com.alibaba.nacos.naming.core;
 
+import com.alibaba.nacos.api.exception.NacosException;
 import com.alibaba.nacos.common.utils.JacksonUtils;
 import com.alibaba.nacos.naming.healthcheck.RsInfo;
 import org.junit.Before;
@@ -131,4 +132,10 @@ public class InstanceTest {
         RsInfo info1 = JacksonUtils.toObj(json, RsInfo.class);
         System.out.println(info1);
     }
+    
+    @Test(expected = NacosException.class)
+    public void testIpValidate() throws NacosException {
+        Instance instance1 = new Instance("192.168.1.3d", 8080);
+        instance1.validate();
+    }
 }
