diff --git a/test/config-test/src/test/java/com/alibaba/nacos/test/base/BaseClusterTest.java b/test/config-test/src/test/java/com/alibaba/nacos/test/base/BaseClusterTest.java
index 3d465ca..f52bc00 100644
--- a/test/config-test/src/test/java/com/alibaba/nacos/test/base/BaseClusterTest.java
+++ b/test/config-test/src/test/java/com/alibaba/nacos/test/base/BaseClusterTest.java
@@ -24,8 +24,8 @@ import com.alibaba.nacos.common.http.client.NacosRestTemplate;
 import com.alibaba.nacos.common.notify.Event;
 import com.alibaba.nacos.common.notify.NotifyCenter;
 import com.alibaba.nacos.common.notify.listener.Subscriber;
-import com.alibaba.nacos.config.server.model.event.RaftDbErrorEvent;
-import com.alibaba.nacos.config.server.service.repository.embedded.DistributedDatabaseOperateImpl;
+import com.alibaba.nacos.persistence.model.event.RaftDbErrorEvent;
+import com.alibaba.nacos.core.persistence.DistributedDatabaseOperateImpl;
 import com.alibaba.nacos.consistency.ProtocolMetaData;
 import com.alibaba.nacos.consistency.cp.CPProtocol;
 import com.alibaba.nacos.consistency.cp.MetadataKey;
diff --git a/test/config-test/src/test/java/com/alibaba/nacos/test/config/ConfigDerbyImport_CITCase.java b/test/config-test/src/test/java/com/alibaba/nacos/test/config/ConfigDerbyImport_CITCase.java
index d0a3396..9930f24 100644
--- a/test/config-test/src/test/java/com/alibaba/nacos/test/config/ConfigDerbyImport_CITCase.java
+++ b/test/config-test/src/test/java/com/alibaba/nacos/test/config/ConfigDerbyImport_CITCase.java
@@ -21,9 +21,9 @@ import com.alibaba.nacos.Nacos;
 import com.alibaba.nacos.common.model.RestResult;
 import com.alibaba.nacos.common.utils.ByteUtils;
 import com.alibaba.nacos.config.server.model.ConfigInfo;
-import com.alibaba.nacos.config.server.service.datasource.DynamicDataSource;
+import com.alibaba.nacos.persistence.datasource.DynamicDataSource;
 import com.alibaba.nacos.config.server.service.repository.PersistService;
-import com.alibaba.nacos.config.server.service.repository.embedded.DatabaseOperate;
+import com.alibaba.nacos.persistence.repository.embedded.operate.DatabaseOperate;
 import com.alibaba.nacos.sys.utils.DiskUtils;
 import com.alibaba.nacos.test.base.ConfigCleanUtils;
 import org.junit.Assert;
diff --git a/test/config-test/src/test/java/com/alibaba/nacos/test/config/ConfigDerbyRaft_DITCase.java b/test/config-test/src/test/java/com/alibaba/nacos/test/config/ConfigDerbyRaft_DITCase.java
index f448190..8b6c301 100644
--- a/test/config-test/src/test/java/com/alibaba/nacos/test/config/ConfigDerbyRaft_DITCase.java
+++ b/test/config-test/src/test/java/com/alibaba/nacos/test/config/ConfigDerbyRaft_DITCase.java
@@ -24,7 +24,7 @@ import com.alibaba.nacos.common.utils.ThreadUtils;
 import com.alibaba.nacos.common.http.param.Header;
 import com.alibaba.nacos.common.http.param.Query;
 import com.alibaba.nacos.common.model.RestResult;
-import com.alibaba.nacos.config.server.model.event.RaftDbErrorEvent;
+import com.alibaba.nacos.persistence.model.event.RaftDbErrorEvent;
 import com.alibaba.nacos.config.server.model.event.RaftDbErrorRecoverEvent;
 import com.alibaba.nacos.config.server.service.repository.embedded.EmbeddedStoragePersistServiceImpl;
 import com.alibaba.nacos.config.server.service.repository.PersistService;
@@ -32,6 +32,7 @@ import com.alibaba.nacos.consistency.cp.CPProtocol;
 import com.alibaba.nacos.core.distributed.id.IdGeneratorManager;
 import com.alibaba.nacos.core.distributed.raft.utils.JRaftConstants;
 import com.alibaba.nacos.core.utils.GenericType;
+import com.alibaba.nacos.persistence.constants.PersistenceConstant;
 import com.alibaba.nacos.sys.utils.InetUtils;
 
 import com.alibaba.nacos.test.base.BaseClusterTest;
@@ -347,7 +348,7 @@ public class ConfigDerbyRaft_DITCase extends BaseClusterTest {
         CPProtocol protocol8 = context8.getBean(CPProtocol.class);
         CPProtocol protocol9 = context9.getBean(CPProtocol.class);
         
-        final String configGroup = com.alibaba.nacos.config.server.constant.Constants.CONFIG_MODEL_RAFT_GROUP;
+        final String configGroup = PersistenceConstant.CONFIG_MODEL_RAFT_GROUP;
         long preId = -1L;
         long currentId = -1L;
         
diff --git a/test/config-test/src/test/java/com/alibaba/nacos/test/config/EmbeddedStorageContextUtils_CITCase.java b/test/config-test/src/test/java/com/alibaba/nacos/test/config/EmbeddedStorageContextUtils_CITCase.java
index 4e2296e..5c8e46e 100644
--- a/test/config-test/src/test/java/com/alibaba/nacos/test/config/EmbeddedStorageContextUtils_CITCase.java
+++ b/test/config-test/src/test/java/com/alibaba/nacos/test/config/EmbeddedStorageContextUtils_CITCase.java
@@ -16,8 +16,8 @@
 
 package com.alibaba.nacos.test.config;
 
-import com.alibaba.nacos.config.server.service.sql.EmbeddedStorageContextUtils;
-import com.alibaba.nacos.config.server.service.sql.ModifyRequest;
+import com.alibaba.nacos.persistence.repository.embedded.EmbeddedStorageContextHolder;
+import com.alibaba.nacos.persistence.repository.embedded.sql.ModifyRequest;
 import com.alibaba.nacos.test.base.ConfigCleanUtils;
 import org.junit.AfterClass;
 import org.junit.Assert;
@@ -50,10 +50,10 @@ public class EmbeddedStorageContextUtils_CITCase {
             final int j = i;
             service.submit(() -> {
                 try {
-                    EmbeddedStorageContextUtils.addSqlContext("test_" + j, j);
-                    EmbeddedStorageContextUtils.addSqlContext("test_" + j * 10, j);
+                    EmbeddedStorageContextHolder.addSqlContext("test_" + j, j);
+                    EmbeddedStorageContextHolder.addSqlContext("test_" + j * 10, j);
                     
-                    List<ModifyRequest> list = EmbeddedStorageContextUtils.getCurrentSqlContext();
+                    List<ModifyRequest> list = EmbeddedStorageContextHolder.getCurrentSqlContext();
                     System.out.println(list);
                     Assert.assertEquals("test_" + j, list.get(0).getSql());
                     Assert.assertEquals("test_" + j * 10, list.get(0).getSql());
