diff --git a/test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/ITContainers.java b/test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/ITContainers.java
index ed64144..cc033c9 100644
--- a/test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/ITContainers.java
+++ b/test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/ITContainers.java
@@ -110,5 +110,6 @@ public final class ITContainers implements Startable {
     public void stop() {
         embeddedContainers.forEach(Startable::close);
         dockerContainers.forEach(Startable::close);
+        network.close();
     }
 }
