diff --git a/kernel/sql-parser/distsql/handler/src/test/java/org/apache/shardingsphere/parser/distsql/handler/query/ShowSQLParserRuleExecutorTest.java b/kernel/sql-parser/distsql/handler/src/test/java/org/apache/shardingsphere/parser/distsql/handler/query/ShowSQLParserRuleExecutorTest.java
index e9e632e..873ba56 100644
--- a/kernel/sql-parser/distsql/handler/src/test/java/org/apache/shardingsphere/parser/distsql/handler/query/ShowSQLParserRuleExecutorTest.java
+++ b/kernel/sql-parser/distsql/handler/src/test/java/org/apache/shardingsphere/parser/distsql/handler/query/ShowSQLParserRuleExecutorTest.java
@@ -59,7 +59,7 @@ class ShowSQLParserRuleExecutorTest {
         Collection<String> columns = executor.getColumnNames();
         assertThat(columns.size(), is(3));
         Iterator<String> iterator = columns.iterator();
-        assertThat(iterator.next(), is("sql_comment_parse_enable"));
+        assertThat(iterator.next(), is("sql_comment_parse_enabled"));
         assertThat(iterator.next(), is("parse_tree_cache"));
         assertThat(iterator.next(), is("sql_statement_cache"));
     }
