diff --git a/store/src/test/java/org/apache/rocketmq/store/timer/TimerMessageStoreTest.java b/store/src/test/java/org/apache/rocketmq/store/timer/TimerMessageStoreTest.java
index 3023f3e..d8e4b1c 100644
--- a/store/src/test/java/org/apache/rocketmq/store/timer/TimerMessageStoreTest.java
+++ b/store/src/test/java/org/apache/rocketmq/store/timer/TimerMessageStoreTest.java
@@ -261,6 +261,7 @@ public class TimerMessageStoreTest {
     public void testPutExpiredTimerMessage() throws Exception {
         // Skip on Mac to make CI pass
         Assume.assumeFalse(MixAll.isMac());
+        Assume.assumeFalse(MixAll.isWindows());
 
         String topic = "TimerTest_testPutExpiredTimerMessage";
 
