diff --git a/test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java b/test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java
index fc599d6..fea6d96 100644
--- a/test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java
+++ b/test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java
@@ -345,7 +345,7 @@ public class StaticTopicIT extends BaseConf {
         int msgEachQueue = 10;
         //create static topic
         {
-            Set<String> targetBrokers = ImmutableSet.of(BROKER2_NAME);
+            Set<String> targetBrokers = ImmutableSet.of(BROKER1_NAME);
             MQAdminTestUtils.createStaticTopic(topic, queueNum, targetBrokers, defaultMQAdminExt);
             sendMessagesAndCheck(producer, targetBrokers, topic, queueNum, msgEachQueue, 0);
             consumeMessagesAndCheck(producer, consumer, topic, queueNum, msgEachQueue, 0, 1);
@@ -409,7 +409,7 @@ public class StaticTopicIT extends BaseConf {
 
         //remapping to broker2Name
         {
-            Set<String> targetBrokers = ImmutableSet.of(BROKER3_NAME);
+            Set<String> targetBrokers = ImmutableSet.of(BROKER2_NAME);
             MQAdminTestUtils.remappingStaticTopic(topic, targetBrokers, defaultMQAdminExt);
             //leave the time to refresh the metadata
             awaitRefreshStaticTopicMetadata(3000, topic, producer.getProducer(), null, defaultMQAdminExt);
@@ -425,7 +425,7 @@ public class StaticTopicIT extends BaseConf {
             sendMessagesAndCheck(producer, targetBrokers, topic, queueNum, msgEachQueue, 2 * TopicQueueMappingUtils.DEFAULT_BLOCK_SEQ_SIZE);
         }
 
-        // 1 -> 2 -> 3, currently 1 should not has any mappings
+        // 1 -> 2 -> 3, currently 1 should not have any mappings
 
         {
             for (int i = 0; i < 10; i++) {
