diff --git a/test/it/optimizer/src/test/java/org/apache/shardingsphere/test/it/optimize/SQLNodeConverterEngineIT.java b/test/it/optimizer/src/test/java/org/apache/shardingsphere/test/it/optimize/SQLNodeConverterEngineIT.java
index 117ca10..3b2f88a 100644
--- a/test/it/optimizer/src/test/java/org/apache/shardingsphere/test/it/optimize/SQLNodeConverterEngineIT.java
+++ b/test/it/optimizer/src/test/java/org/apache/shardingsphere/test/it/optimize/SQLNodeConverterEngineIT.java
@@ -206,6 +206,8 @@ class SQLNodeConverterEngineIT {
             result.add("select_where_with_bit_expr_with_plus_interval");
             result.add("select_where_with_bit_expr_with_minus_interval");
             result.add("select_where_with_predicate_with_in_subquery");
+            result.add("select_where_with_boolean_primary_with_is");
+            result.add("select_where_with_boolean_primary_with_is_not");
             return result;
         }
         // CHECKSTYLE:ON
