diff --git a/remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RegisterBrokerBodyTest.java b/remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RegisterBrokerBodyTest.java
index 0d32a80..e63552f 100644
--- a/remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RegisterBrokerBodyTest.java
+++ b/remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RegisterBrokerBodyTest.java
@@ -20,6 +20,8 @@ package org.apache.rocketmq.remoting.protocol;
 import java.io.IOException;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+
+import org.apache.rocketmq.common.MQVersion;
 import org.apache.rocketmq.common.TopicConfig;
 import org.apache.rocketmq.remoting.protocol.body.RegisterBrokerBody;
 import org.apache.rocketmq.remoting.protocol.body.TopicConfigAndMappingSerializeWrapper;
@@ -43,7 +45,7 @@ public class RegisterBrokerBodyTest {
 
         byte[] compareEncode = registerBrokerBody.encode(true);
         byte[] encode2 = registerBrokerBody.encode(false);
-        RegisterBrokerBody decodeRegisterBrokerBody = RegisterBrokerBody.decode(compareEncode, true);
+        RegisterBrokerBody decodeRegisterBrokerBody = RegisterBrokerBody.decode(compareEncode, true, MQVersion.Version.V5_0_0);
 
         assertEquals(registerBrokerBody.getTopicConfigSerializeWrapper().getTopicConfigTable().size(), decodeRegisterBrokerBody.getTopicConfigSerializeWrapper().getTopicConfigTable().size());
 
