diff --git a/broker/src/test/java/org/apache/rocketmq/broker/failover/EscapeBridgeTest.java b/broker/src/test/java/org/apache/rocketmq/broker/failover/EscapeBridgeTest.java
index 685e85d..a51e542 100644
--- a/broker/src/test/java/org/apache/rocketmq/broker/failover/EscapeBridgeTest.java
+++ b/broker/src/test/java/org/apache/rocketmq/broker/failover/EscapeBridgeTest.java
@@ -64,8 +64,6 @@ public class EscapeBridgeTest {
 
     private static final String BROKER_NAME = "broker_a";
 
-    private static final String NAMESERVER_ADDR = "127.0.0.1:9876";
-
     private static final String TEST_TOPIC = "TEST_TOPIC";
 
     private static final int DEFAULT_QUEUE_ID = 0;
@@ -81,7 +79,6 @@ public class EscapeBridgeTest {
         escapeBridge = new EscapeBridge(brokerController);
         messageExtBrokerInner = new MessageExtBrokerInner();
         when(brokerController.getMessageStore()).thenReturn(defaultMessageStore);
-        when(brokerController.getNameServerList()).thenReturn(NAMESERVER_ADDR);
         brokerConfig.setEnableSlaveActingMaster(true);
         brokerConfig.setEnableRemoteEscape(true);
         escapeBridge.start();
diff --git a/broker/src/test/java/org/apache/rocketmq/broker/processor/QueryAssignmentProcessorTest.java b/broker/src/test/java/org/apache/rocketmq/broker/processor/QueryAssignmentProcessorTest.java
index 347a240..0367e3a 100644
--- a/broker/src/test/java/org/apache/rocketmq/broker/processor/QueryAssignmentProcessorTest.java
+++ b/broker/src/test/java/org/apache/rocketmq/broker/processor/QueryAssignmentProcessorTest.java
@@ -23,7 +23,7 @@ import java.util.ArrayList;
 import java.util.List;
 import org.apache.rocketmq.broker.BrokerController;
 import org.apache.rocketmq.broker.client.ClientChannelInfo;
-import org.apache.rocketmq.broker.loadbalance.AssignmentManager;
+import org.apache.rocketmq.broker.topic.TopicRouteInfoManager;
 import org.apache.rocketmq.client.consumer.AllocateMessageQueueStrategy;
 import org.apache.rocketmq.client.consumer.rebalance.AllocateMessageQueueAveragely;
 import org.apache.rocketmq.client.consumer.rebalance.AllocateMessageQueueAveragelyByCircle;
@@ -66,7 +66,7 @@ public class QueryAssignmentProcessorTest {
     private BrokerController brokerController = new BrokerController(new BrokerConfig(), new NettyServerConfig(), new NettyClientConfig(), new MessageStoreConfig());
 
     @Mock
-    private AssignmentManager assignmentManager;
+    private TopicRouteInfoManager topicRouteInfoManager;
     @Mock
     private ChannelHandlerContext handlerContext;
     @Mock
@@ -84,8 +84,8 @@ public class QueryAssignmentProcessorTest {
     public void init() throws IllegalAccessException, NoSuchFieldException {
         clientInfo = new ClientChannelInfo(channel, "127.0.0.1", LanguageCode.JAVA, 0);
         brokerController.setMessageStore(messageStore);
-        doReturn(assignmentManager).when(brokerController).getAssignmentManager();
-        when(assignmentManager.getTopicSubscribeInfo(topic)).thenReturn(ImmutableSet.of(new MessageQueue(topic, "broker-1", 0), new MessageQueue(topic, "broker-2", 1)));
+        doReturn(topicRouteInfoManager).when(brokerController).getTopicRouteInfoManager();
+        when(topicRouteInfoManager.getTopicSubscribeInfo(topic)).thenReturn(ImmutableSet.of(new MessageQueue(topic, "broker-1", 0), new MessageQueue(topic, "broker-2", 1)));
         queryAssignmentProcessor = new QueryAssignmentProcessor(brokerController);
         brokerController.getTopicConfigManager().getTopicConfigTable().put(topic, new TopicConfig());
         ConsumerData consumerData = createConsumerData(group, topic);
