diff --git a/client/src/test/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumerTest.java b/client/src/test/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumerTest.java
index 9b3d4b9..5fc4df8 100644
--- a/client/src/test/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumerTest.java
+++ b/client/src/test/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumerTest.java
@@ -194,7 +194,7 @@ public class DefaultLitePullConsumerTest {
         //commit offset 1
         Map<MessageQueue, Long> commitOffset = new HashMap<>();
         commitOffset.put(messageQueue, 1L);
-        litePullConsumer.commitSync(commitOffset, true);
+        litePullConsumer.commit(commitOffset, true);
 
         assertThat(litePullConsumer.committed(messageQueue)).isEqualTo(1);
     }
