diff --git a/broker/src/test/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessorTest.java b/broker/src/test/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessorTest.java
index e0436f1..3d70247 100644
--- a/broker/src/test/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessorTest.java
+++ b/broker/src/test/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessorTest.java
@@ -157,6 +157,7 @@ public class AdminBrokerProcessorTest {
         topic = "FooBar" + System.nanoTime();
 
         brokerController.getTopicConfigManager().getTopicConfigTable().put(topic, new TopicConfig(topic));
+        brokerController.getMessageStoreConfig().setTimerWheelEnable(false);
     }
 
     @Test
@@ -514,7 +515,6 @@ public class AdminBrokerProcessorTest {
         assertThat(response.getCode()).isEqualTo(ResponseCode.SUCCESS);
     }
 
-
     @Test
     public void testGetTopicConfig() throws Exception {
         String topic = "foobar";
