diff --git a/kernel/metadata/core/src/test/java/org/apache/shardingsphere/metadata/persist/service/DatabaseMetaDataPersistServiceTest.java b/kernel/metadata/core/src/test/java/org/apache/shardingsphere/metadata/persist/service/DatabaseMetaDataPersistServiceTest.java
index 25f10c5..1f5fd39 100644
--- a/kernel/metadata/core/src/test/java/org/apache/shardingsphere/metadata/persist/service/DatabaseMetaDataPersistServiceTest.java
+++ b/kernel/metadata/core/src/test/java/org/apache/shardingsphere/metadata/persist/service/DatabaseMetaDataPersistServiceTest.java
@@ -127,11 +127,11 @@ class DatabaseMetaDataPersistServiceTest {
         assertThat(schema.size(), is(1));
         assertTrue(databaseMetaDataPersistService.loadSchemas("test").isEmpty());
         assertThat(schema.get("foo_schema").getAllTableNames(), is(Collections.singleton("t_order")));
-        assertThat(schema.get("foo_schema").getTable("t_order").getIndexes().size(), is(1));
-        assertThat(schema.get("foo_schema").getTable("t_order").getIndexes().iterator().next().getName(), is("PRIMARY"));
+        assertThat(schema.get("foo_schema").getTable("t_order").getIndexValues().size(), is(1));
+        assertThat(schema.get("foo_schema").getTable("t_order").getIndexValues().iterator().next().getName(), is("PRIMARY"));
         assertThat(schema.get("foo_schema").getAllColumnNames("t_order").size(), is(1));
-        assertThat(schema.get("foo_schema").getTable("t_order").getColumns().size(), is(1));
-        assertThat(schema.get("foo_schema").getTable("t_order").getColumns().iterator().next().getName(), is("id"));
+        assertThat(schema.get("foo_schema").getTable("t_order").getColumnValues().size(), is(1));
+        assertThat(schema.get("foo_schema").getTable("t_order").getColumnValues().iterator().next().getName(), is("id"));
     }
     
     @SneakyThrows({IOException.class, URISyntaxException.class})
diff --git a/kernel/metadata/core/src/test/java/org/apache/shardingsphere/metadata/persist/service/schema/TableMetaDataPersistServiceTest.java b/kernel/metadata/core/src/test/java/org/apache/shardingsphere/metadata/persist/service/schema/TableMetaDataPersistServiceTest.java
index 03ee83c..401fe57 100644
--- a/kernel/metadata/core/src/test/java/org/apache/shardingsphere/metadata/persist/service/schema/TableMetaDataPersistServiceTest.java
+++ b/kernel/metadata/core/src/test/java/org/apache/shardingsphere/metadata/persist/service/schema/TableMetaDataPersistServiceTest.java
@@ -58,10 +58,10 @@ class TableMetaDataPersistServiceTest {
         when(repository.getDirectly("/metadata/foo_db/schemas/foo_schema/tables/t_order")).thenReturn(readYAML());
         Map<String, ShardingSphereTable> tables = tableMetaDataPersistService.load("foo_db", "foo_schema");
         assertThat(tables.size(), is(1));
-        assertThat(tables.get("t_order").getIndexes().size(), is(1));
-        assertThat(tables.get("t_order").getIndexes().iterator().next().getName(), is("PRIMARY"));
-        assertThat(tables.get("t_order").getColumns().size(), is(1));
-        assertThat(tables.get("t_order").getColumns().iterator().next().getName(), is("id"));
+        assertThat(tables.get("t_order").getIndexValues().size(), is(1));
+        assertThat(tables.get("t_order").getIndexValues().iterator().next().getName(), is("PRIMARY"));
+        assertThat(tables.get("t_order").getColumnValues().size(), is(1));
+        assertThat(tables.get("t_order").getColumnValues().iterator().next().getName(), is("id"));
     }
     
     @Test
