diff --git a/broker/src/test/java/org/apache/rocketmq/broker/BrokerPathConfigHelperTest.java b/broker/src/test/java/org/apache/rocketmq/broker/BrokerPathConfigHelperTest.java
index 01e7c36..3b26054 100644
--- a/broker/src/test/java/org/apache/rocketmq/broker/BrokerPathConfigHelperTest.java
+++ b/broker/src/test/java/org/apache/rocketmq/broker/BrokerPathConfigHelperTest.java
@@ -17,6 +17,7 @@
 
 package org.apache.rocketmq.broker;
 
+import java.io.File;
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
@@ -25,18 +26,17 @@ public class BrokerPathConfigHelperTest {
 
     @Test
     public void testGetLmqConsumerOffsetPath() {
-        String lmqConsumerOffsetPath = BrokerPathConfigHelper.getLmqConsumerOffsetPath("/home/admin/store");
-        assertEquals("/home/admin/store/config/lmqConsumerOffset.json", lmqConsumerOffsetPath);
+        String lmqConsumerOffsetPath = BrokerPathConfigHelper.getLmqConsumerOffsetPath("/home/admin/store".replace("/", File.separator));
+        assertEquals("/home/admin/store/config/lmqConsumerOffset.json".replace("/", File.separator), lmqConsumerOffsetPath);
 
+        String consumerOffsetPath = BrokerPathConfigHelper.getConsumerOffsetPath("/home/admin/store".replace("/", File.separator));
+        assertEquals("/home/admin/store/config/consumerOffset.json".replace("/", File.separator), consumerOffsetPath);
 
-        String consumerOffsetPath = BrokerPathConfigHelper.getConsumerOffsetPath("/home/admin/store");
-        assertEquals("/home/admin/store/config/consumerOffset.json", consumerOffsetPath);
+        String topicConfigPath = BrokerPathConfigHelper.getTopicConfigPath("/home/admin/store".replace("/", File.separator));
+        assertEquals("/home/admin/store/config/topics.json".replace("/", File.separator), topicConfigPath);
 
-        String topicConfigPath = BrokerPathConfigHelper.getTopicConfigPath("/home/admin/store");
-        assertEquals("/home/admin/store/config/topics.json", topicConfigPath);
-
-        String subscriptionGroupPath = BrokerPathConfigHelper.getSubscriptionGroupPath("/home/admin/store");
-        assertEquals("/home/admin/store/config/subscriptionGroup.json", subscriptionGroupPath);
+        String subscriptionGroupPath = BrokerPathConfigHelper.getSubscriptionGroupPath("/home/admin/store".replace("/", File.separator));
+        assertEquals("/home/admin/store/config/subscriptionGroup.json".replace("/", File.separator), subscriptionGroupPath);
 
     }
 }
\ No newline at end of file
