diff --git a/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassLoaderResourceLoaderTest.java b/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassLoaderResourceLoaderTest.java
index 4592103..efcfdbf 100644
--- a/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassLoaderResourceLoaderTest.java
+++ b/dubbo-common/src/test/java/org/apache/dubbo/common/utils/ClassLoaderResourceLoaderTest.java
@@ -34,7 +34,7 @@ import java.util.Set;
 public class ClassLoaderResourceLoaderTest {
 
     @Test
-    public void test() {
+    public void test() throws InterruptedException {
         DubboInternalLoadingStrategy dubboInternalLoadingStrategy = new DubboInternalLoadingStrategy();
         String directory = dubboInternalLoadingStrategy.directory();
         String type = FooAppProvider.class.getName();
