diff --git a/proxy/src/test/java/org/apache/rocketmq/proxy/config/MetricCollectorModeTest.java b/proxy/src/test/java/org/apache/rocketmq/proxy/config/MetricCollectorModeTest.java
index 59aa49f..60c754e 100644
--- a/proxy/src/test/java/org/apache/rocketmq/proxy/config/MetricCollectorModeTest.java
+++ b/proxy/src/test/java/org/apache/rocketmq/proxy/config/MetricCollectorModeTest.java
@@ -24,9 +24,13 @@ public class MetricCollectorModeTest {
 
     @Test
     public void testGetEnumByOrdinal() {
-        Assert.assertEquals(MetricCollectorMode.OFF, MetricCollectorMode.getEnumByOrdinal(0));
-        Assert.assertEquals(MetricCollectorMode.ON, MetricCollectorMode.getEnumByOrdinal(1));
-        Assert.assertEquals(MetricCollectorMode.PROXY, MetricCollectorMode.getEnumByOrdinal(2));
+        Assert.assertEquals(MetricCollectorMode.OFF, MetricCollectorMode.getEnumByString("off"));
+        Assert.assertEquals(MetricCollectorMode.ON, MetricCollectorMode.getEnumByString("on"));
+        Assert.assertEquals(MetricCollectorMode.PROXY, MetricCollectorMode.getEnumByString("proxy"));
+
+        Assert.assertEquals(MetricCollectorMode.OFF, MetricCollectorMode.getEnumByString("OFF"));
+        Assert.assertEquals(MetricCollectorMode.ON, MetricCollectorMode.getEnumByString("ON"));
+        Assert.assertEquals(MetricCollectorMode.PROXY, MetricCollectorMode.getEnumByString("PROXY"));
     }
 
 }
\ No newline at end of file
