diff --git a/broker/src/test/java/org/apache/rocketmq/broker/processor/PopBufferMergeServiceTest.java b/broker/src/test/java/org/apache/rocketmq/broker/processor/PopBufferMergeServiceTest.java
index 1a53b94..acc7a3d 100644
--- a/broker/src/test/java/org/apache/rocketmq/broker/processor/PopBufferMergeServiceTest.java
+++ b/broker/src/test/java/org/apache/rocketmq/broker/processor/PopBufferMergeServiceTest.java
@@ -100,7 +100,7 @@ public class PopBufferMergeServiceTest {
         ck.setCId(group);
         ck.setTopic(topic);
         int queueId = 0;
-        ck.setQueueId((byte) queueId);
+        ck.setQueueId(queueId);
 
         int reviveQid = 0;
         long nextBeginOffset = 101L;
diff --git a/broker/src/test/java/org/apache/rocketmq/broker/processor/PopInflightMessageCounterTest.java b/broker/src/test/java/org/apache/rocketmq/broker/processor/PopInflightMessageCounterTest.java
index 3b50919..4e83ac7 100644
--- a/broker/src/test/java/org/apache/rocketmq/broker/processor/PopInflightMessageCounterTest.java
+++ b/broker/src/test/java/org/apache/rocketmq/broker/processor/PopInflightMessageCounterTest.java
@@ -53,7 +53,7 @@ public class PopInflightMessageCounterTest {
         PopCheckPoint popCheckPoint = new PopCheckPoint();
         popCheckPoint.setTopic(topic);
         popCheckPoint.setCId(group);
-        popCheckPoint.setQueueId((byte) 0);
+        popCheckPoint.setQueueId(0);
         popCheckPoint.setPopTime(System.currentTimeMillis());
 
         counter.decrementInFlightMessageNum(popCheckPoint);
diff --git a/broker/src/test/java/org/apache/rocketmq/broker/processor/PopReviveServiceTest.java b/broker/src/test/java/org/apache/rocketmq/broker/processor/PopReviveServiceTest.java
index 89ffed7..1c3a0cd 100644
--- a/broker/src/test/java/org/apache/rocketmq/broker/processor/PopReviveServiceTest.java
+++ b/broker/src/test/java/org/apache/rocketmq/broker/processor/PopReviveServiceTest.java
@@ -208,7 +208,7 @@ public class PopReviveServiceTest {
         PopCheckPoint ck = new PopCheckPoint();
         ck.setStartOffset(startOffset);
         ck.setPopTime(popTime);
-        ck.setQueueId((byte) 0);
+        ck.setQueueId(0);
         ck.setCId(GROUP);
         ck.setTopic(TOPIC);
         ck.setNum((byte) 1);
