diff --git a/logging/src/test/java/org/apache/rocketmq/logging/InnerLoggerFactoryTest.java b/logging/src/test/java/org/apache/rocketmq/logging/InnerLoggerFactoryTest.java
index c47dba6..2faaabc 100644
--- a/logging/src/test/java/org/apache/rocketmq/logging/InnerLoggerFactoryTest.java
+++ b/logging/src/test/java/org/apache/rocketmq/logging/InnerLoggerFactoryTest.java
@@ -82,7 +82,6 @@ public class InnerLoggerFactoryTest extends BasicLoggerTest {
         logger3.debug("debug {}", "hahahah");
 
         String content = new String(byteArrayOutputStream.toByteArray());
-        System.out.println(content);
 
         Assert.assertTrue(content.contains("InnerLoggerFactoryTest"));
         Assert.assertTrue(content.contains("info"));
diff --git a/logging/src/test/java/org/apache/rocketmq/logging/inner/AppenderTest.java b/logging/src/test/java/org/apache/rocketmq/logging/inner/AppenderTest.java
index 37ff8bd..cd3d0aa 100644
--- a/logging/src/test/java/org/apache/rocketmq/logging/inner/AppenderTest.java
+++ b/logging/src/test/java/org/apache/rocketmq/logging/inner/AppenderTest.java
@@ -91,8 +91,6 @@ public class AppenderTest extends BasicLoggerTest {
 
         String content = readFile(file);
 
-        System.out.println(content);
-
         Assert.assertTrue(content.contains("info"));
         Assert.assertTrue(content.contains("RuntimeException"));
         Assert.assertTrue(!content.contains("debug"));
diff --git a/logging/src/test/java/org/apache/rocketmq/logging/inner/LayoutTest.java b/logging/src/test/java/org/apache/rocketmq/logging/inner/LayoutTest.java
index 66ef18e..c48be1d 100644
--- a/logging/src/test/java/org/apache/rocketmq/logging/inner/LayoutTest.java
+++ b/logging/src/test/java/org/apache/rocketmq/logging/inner/LayoutTest.java
@@ -48,7 +48,5 @@ public class LayoutTest extends BasicLoggerTest {
         LoggingEvent loggingEvent = new LoggingEvent(Logger.class.getName(), logger, org.apache.rocketmq.logging.inner.Level.INFO,
             "junit test error", null);
         String format = innerLayout.format(loggingEvent);
-
-        System.out.println(format);
     }
 }
diff --git a/logging/src/test/java/org/apache/rocketmq/logging/inner/LevelTest.java b/logging/src/test/java/org/apache/rocketmq/logging/inner/LevelTest.java
index 21667e1..297523a 100644
--- a/logging/src/test/java/org/apache/rocketmq/logging/inner/LevelTest.java
+++ b/logging/src/test/java/org/apache/rocketmq/logging/inner/LevelTest.java
@@ -30,8 +30,8 @@ public class LevelTest {
     }
 
     @Test
-    public void loggerLevel(){
+    public void loggerLevel() {
         Level level = Logger.getRootLogger().getLevel();
-        Assert.assertTrue(level!=null);
+        Assert.assertTrue(level != null);
     }
 }
diff --git a/logging/src/test/java/org/apache/rocketmq/logging/inner/LoggerTest.java b/logging/src/test/java/org/apache/rocketmq/logging/inner/LoggerTest.java
index 4e738e2..904c632 100644
--- a/logging/src/test/java/org/apache/rocketmq/logging/inner/LoggerTest.java
+++ b/logging/src/test/java/org/apache/rocketmq/logging/inner/LoggerTest.java
@@ -66,8 +66,6 @@ public class LoggerTest extends BasicLoggerTest {
 
         String result = new String(byteArrayOutputStream.toByteArray());
 
-        System.out.println(result);
-
         Assert.assertTrue(result.contains("info"));
         Assert.assertTrue(result.contains("RuntimeException"));
         Assert.assertTrue(result.contains("WATERMARK"));
@@ -101,8 +99,6 @@ public class LoggerTest extends BasicLoggerTest {
 
         String content = readFile(file);
 
-        System.out.println(content);
-
         Assert.assertTrue(content.contains("info"));
         Assert.assertTrue(content.contains("RuntimeException"));
         Assert.assertTrue(!content.contains("debug"));
diff --git a/logging/src/test/java/org/apache/rocketmq/logging/inner/MessageFormatterTest.java b/logging/src/test/java/org/apache/rocketmq/logging/inner/MessageFormatterTest.java
index 5fa80ad..5d3d868 100644
--- a/logging/src/test/java/org/apache/rocketmq/logging/inner/MessageFormatterTest.java
+++ b/logging/src/test/java/org/apache/rocketmq/logging/inner/MessageFormatterTest.java
@@ -25,7 +25,7 @@ import org.junit.Test;
 public class MessageFormatterTest {
 
     @Test
-    public void formatTest(){
+    public void formatTest() {
         InnerLoggerFactory.FormattingTuple logging = InnerLoggerFactory.MessageFormatter.format("this is {},and {}", "logging", 6546);
         String message = logging.getMessage();
         Assert.assertTrue(message.contains("logging"));
@@ -33,7 +33,6 @@ public class MessageFormatterTest {
         InnerLoggerFactory.FormattingTuple format = InnerLoggerFactory.MessageFormatter.format("cause exception {}", 143545, new RuntimeException());
         String message1 = format.getMessage();
         Throwable throwable = format.getThrowable();
-        System.out.println(message1);
         Assert.assertTrue(throwable != null);
     }
 
