diff --git a/tools/src/test/java/org/apache/rocketmq/tools/command/broker/BrokerStatusSubCommandTest.java b/tools/src/test/java/org/apache/rocketmq/tools/command/broker/BrokerStatusSubCommandTest.java
index d329fb0..c685a06 100644
--- a/tools/src/test/java/org/apache/rocketmq/tools/command/broker/BrokerStatusSubCommandTest.java
+++ b/tools/src/test/java/org/apache/rocketmq/tools/command/broker/BrokerStatusSubCommandTest.java
@@ -39,7 +39,7 @@ public class BrokerStatusSubCommandTest extends ServerResponseMocker {
     public void testExecute() throws SubCommandException {
         BrokerStatusSubCommand cmd = new BrokerStatusSubCommand();
         Options options = ServerUtil.buildCommandlineOptions(new Options());
-        String[] subargs = new String[] {"-b 127.0.0.1:" + listenPort(), "-c default-cluster"};
+        String[] subargs = new String[] {"-b 127.0.0.1:" + listenPort()};
         final CommandLine commandLine =
             ServerUtil.parseCmdLine("mqadmin " + cmd.commandName(), subargs,
                 cmd.buildCommandlineOptions(options), new DefaultParser());
@@ -47,5 +47,4 @@ public class BrokerStatusSubCommandTest extends ServerResponseMocker {
         cmd.execute(commandLine, options, null);
     }
 
-
 }
