diff --git a/infra/common/src/test/java/org/apache/shardingsphere/infra/database/metadata/dialect/OpenGaussDataSourceMetaDataTest.java b/infra/common/src/test/java/org/apache/shardingsphere/infra/database/metadata/dialect/OpenGaussDataSourceMetaDataTest.java
index b978ff8..c081841 100644
--- a/infra/common/src/test/java/org/apache/shardingsphere/infra/database/metadata/dialect/OpenGaussDataSourceMetaDataTest.java
+++ b/infra/common/src/test/java/org/apache/shardingsphere/infra/database/metadata/dialect/OpenGaussDataSourceMetaDataTest.java
@@ -30,7 +30,7 @@ class OpenGaussDataSourceMetaDataTest {
     
     @Test
     void assertNewConstructorWithSimpleJdbcUrl() {
-        PostgreSQLDataSourceMetaData actual = new PostgreSQLDataSourceMetaData("jdbc:openGauss://127.0.0.1/foo_ds");
+        PostgreSQLDataSourceMetaData actual = new PostgreSQLDataSourceMetaData("jdbc:opengauss://127.0.0.1/foo_ds");
         assertThat(actual.getHostname(), is("127.0.0.1"));
         assertThat(actual.getPort(), is(5432));
         assertThat(actual.getCatalog(), is("foo_ds"));
@@ -40,7 +40,7 @@ class OpenGaussDataSourceMetaDataTest {
     
     @Test
     void assertNewConstructorWithComplexJdbcUrl() {
-        PostgreSQLDataSourceMetaData actual = new PostgreSQLDataSourceMetaData("jdbc:openGauss://127.0.0.1:9999,127.0.0.2:9999,127.0.0.3:9999/foo_ds?targetServerType=master");
+        PostgreSQLDataSourceMetaData actual = new PostgreSQLDataSourceMetaData("jdbc:opengauss://127.0.0.1:9999,127.0.0.2:9999,127.0.0.3:9999/foo_ds?targetServerType=master");
         assertThat(actual.getHostname(), is("127.0.0.1"));
         assertThat(actual.getPort(), is(9999));
         assertThat(actual.getCatalog(), is("foo_ds"));
@@ -51,6 +51,6 @@ class OpenGaussDataSourceMetaDataTest {
     
     @Test
     void assertNewConstructorFailure() {
-        assertThrows(UnrecognizedDatabaseURLException.class, () -> new PostgreSQLDataSourceMetaData("jdbc:openGauss:xxxxxxxx"));
+        assertThrows(UnrecognizedDatabaseURLException.class, () -> new PostgreSQLDataSourceMetaData("jdbc:opengauss:xxxxxxxx"));
     }
 }
