diff --git a/acl/src/test/java/org/apache/rocketmq/acl/plain/PlainAccessValidatorTest.java b/acl/src/test/java/org/apache/rocketmq/acl/plain/PlainAccessValidatorTest.java
index ddd4a97..739f4d2 100644
--- a/acl/src/test/java/org/apache/rocketmq/acl/plain/PlainAccessValidatorTest.java
+++ b/acl/src/test/java/org/apache/rocketmq/acl/plain/PlainAccessValidatorTest.java
@@ -16,18 +16,6 @@
  */
 package org.apache.rocketmq.acl.plain;
 
-import java.io.File;
-import java.io.FileWriter;
-import java.io.IOException;
-import java.lang.reflect.Field;
-import java.nio.ByteBuffer;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
 import org.apache.rocketmq.acl.common.AclClientRPCHook;
 import org.apache.rocketmq.acl.common.AclConstants;
 import org.apache.rocketmq.acl.common.AclException;
@@ -56,6 +44,19 @@ import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
 public class PlainAccessValidatorTest {
 
     private PlainAccessValidator plainAccessValidator;
@@ -906,7 +907,8 @@ public class PlainAccessValidatorTest {
 
     @Test
     public void deleteAccessAclToEmptyTest() {
-        System.setProperty("rocketmq.acl.plain.file", "/conf/empty.yml");
+        final String bakAclFileProp = System.getProperty("rocketmq.acl.plain.file");
+        System.setProperty("rocketmq.acl.plain.file", "conf/empty.yml");
         PlainAccessConfig plainAccessConfig = new PlainAccessConfig();
         plainAccessConfig.setAccessKey("deleteAccessAclToEmpty");
         plainAccessConfig.setSecretKey("12345678");
@@ -914,7 +916,11 @@ public class PlainAccessValidatorTest {
         PlainAccessValidator plainAccessValidator = new PlainAccessValidator();
         plainAccessValidator.updateAccessConfig(plainAccessConfig);
         boolean success = plainAccessValidator.deleteAccessConfig("deleteAccessAclToEmpty");
-        System.setProperty("rocketmq.acl.plain.file", "/conf/plain_acl.yml");
+        if (null != bakAclFileProp) {
+            System.setProperty("rocketmq.acl.plain.file", bakAclFileProp);
+        } else {
+            System.clearProperty("rocketmq.acl.plain.file");
+        }
         Assert.assertTrue(success);
     }
 
