diff --git a/test/src/main/java/org/apache/rocketmq/test/util/FileUtil.java b/test/src/main/java/org/apache/rocketmq/test/util/FileUtil.java
index b6106e0..de11010 100644
--- a/test/src/main/java/org/apache/rocketmq/test/util/FileUtil.java
+++ b/test/src/main/java/org/apache/rocketmq/test/util/FileUtil.java
@@ -24,7 +24,7 @@ import java.util.Map.Entry;
 import java.util.Properties;
 
 public class FileUtil {
-    private static String lineSeperator = System.getProperty("line.separator");
+    private static String lineSeparator = System.getProperty("line.separator");
 
     private String filePath = "";
     private String fileName = "";
@@ -53,7 +53,7 @@ public class FileUtil {
 
     public void appendFile(String content) {
         File file = openFile();
-        String newContent = lineSeperator + content;
+        String newContent = lineSeparator + content;
         writeFile(file, newContent, true);
     }
 
@@ -71,7 +71,7 @@ public class FileUtil {
         StringBuilder sb = new StringBuilder();
         for (Entry<Object, Object> keyEnty : properties.entrySet()) {
             sb.append(keyEnty.getKey()).append("=").append((String) keyEnty.getValue())
-                    .append(lineSeperator);
+                    .append(lineSeparator);
         }
         return sb.toString();
     }
diff --git a/test/src/main/java/org/apache/rocketmq/test/util/VerifyUtils.java b/test/src/main/java/org/apache/rocketmq/test/util/VerifyUtils.java
index 69bfd8f..af4ecca 100644
--- a/test/src/main/java/org/apache/rocketmq/test/util/VerifyUtils.java
+++ b/test/src/main/java/org/apache/rocketmq/test/util/VerifyUtils.java
@@ -40,18 +40,18 @@ public class VerifyUtils {
     public static Collection<Object> getFilterdMessage(Collection<Object> sendMsgs,
         Collection<Object> recvMsgs) {
         Collection<Object> recvMsgsSync = Collections.synchronizedCollection(recvMsgs);
-        Collection<Object> filterdMsgs = new ArrayList<Object>();
+        Collection<Object> filteredMsgs = new ArrayList<Object>();
         int filterNum = 0;
         for (Object msg : recvMsgsSync) {
             if (sendMsgs.contains(msg)) {
-                filterdMsgs.add(msg);
+                filteredMsgs.add(msg);
             } else {
                 filterNum++;
             }
         }
 
-        logger.info(String.format("[%s] messages is filterd!", filterNum));
-        return filterdMsgs;
+        logger.info(String.format("[%s] messages is filtered!", filterNum));
+        return filteredMsgs;
     }
 
     public static int verifyUserProperty(Collection<Object> sendMsgs, Collection<Object> recvMsgs) {
diff --git a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/BaseBroadCastIT.java b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/BaseBroadCastIT.java
index 94efaef..c9ce766 100644
--- a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/BaseBroadCastIT.java
+++ b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/BaseBroadCastIT.java
@@ -47,7 +47,7 @@ public class BaseBroadCastIT extends BaseConf {
         return consumer;
     }
 
-    public void printSeperator() {
+    public void printSeparator() {
         for (int i = 0; i < 3; i++) {
             logger.info(
                 "<<<<<<<<================================================================================>>>>>>>>");
diff --git a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgNotRecvIT.java b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgNotRecvIT.java
index 282a248..f0e42ce 100644
--- a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgNotRecvIT.java
+++ b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgNotRecvIT.java
@@ -38,7 +38,7 @@ public class BroadCastNormalMsgNotRecvIT extends BaseBroadCastIT {
 
     @Before
     public void setUp() {
-        printSeperator();
+        printSeparator();
         topic = initTopic();
         logger.info(String.format("use topic: %s;", topic));
         producer = getProducer(nsAddr, topic);
diff --git a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvCrashIT.java b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvCrashIT.java
index 977e61f..cf61610 100644
--- a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvCrashIT.java
+++ b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvCrashIT.java
@@ -39,7 +39,7 @@ public class BroadCastNormalMsgRecvCrashIT extends BaseBroadCastIT {
 
     @Before
     public void setUp() {
-        printSeperator();
+        printSeparator();
         topic = initTopic();
         logger.info(String.format("use topic: %s;", topic));
         producer = getProducer(nsAddr, topic);
diff --git a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvFailIT.java b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvFailIT.java
index 615c8a5..b4a963a 100644
--- a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvFailIT.java
+++ b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvFailIT.java
@@ -40,7 +40,7 @@ public class BroadCastNormalMsgRecvFailIT extends BaseBroadCastIT {
 
     @Before
     public void setUp() {
-        printSeperator();
+        printSeparator();
         topic = initTopic();
         logger.info(String.format("use topic: %s;", topic));
         producer = getProducer(nsAddr, topic);
diff --git a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvStartLaterIT.java b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvStartLaterIT.java
index 987b4b2..6966faa 100644
--- a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvStartLaterIT.java
+++ b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgRecvStartLaterIT.java
@@ -39,7 +39,7 @@ public class BroadCastNormalMsgRecvStartLaterIT extends BaseBroadCastIT {
 
     @Before
     public void setUp() {
-        printSeperator();
+        printSeparator();
         topic = initTopic();
         logger.info(String.format("use topic: %s;", topic));
         producer = getProducer(nsAddr, topic);
diff --git a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgTwoDiffGroupRecvIT.java b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgTwoDiffGroupRecvIT.java
index 019bcbb..5497d16 100644
--- a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgTwoDiffGroupRecvIT.java
+++ b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/BroadCastNormalMsgTwoDiffGroupRecvIT.java
@@ -39,7 +39,7 @@ public class BroadCastNormalMsgTwoDiffGroupRecvIT extends BaseBroadCastIT {
 
     @Before
     public void setUp() {
-        printSeperator();
+        printSeparator();
         topic = initTopic();
         logger.info(String.format("use topic: %s;", topic));
         producer = getProducer(nsAddr, topic);
diff --git a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/NormalMsgTwoSameGroupConsumerIT.java b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/NormalMsgTwoSameGroupConsumerIT.java
index 4c9e491..7cf0785 100644
--- a/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/NormalMsgTwoSameGroupConsumerIT.java
+++ b/test/src/test/java/org/apache/rocketmq/test/client/consumer/broadcast/normal/NormalMsgTwoSameGroupConsumerIT.java
@@ -39,7 +39,7 @@ public class NormalMsgTwoSameGroupConsumerIT extends BaseBroadCastIT {
 
     @Before
     public void setUp() {
-        printSeperator();
+        printSeparator();
         topic = initTopic();
         logger.info(String.format("use topic: %s;", topic));
         producer = getProducer(nsAddr, topic);
