diff --git a/acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java b/acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java
index d2dabc0..0706b85 100644
--- a/acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java
+++ b/acl/src/test/java/org/apache/rocketmq/acl/plain/PlainPermissionManagerTest.java
@@ -213,6 +213,38 @@ public class PlainPermissionManagerTest {
     }
 
     @Test
+    public void multiFilePathTest() {
+        File file = new File("src/test/resources");
+        System.setProperty("rocketmq.home.dir", file.getAbsolutePath());
+
+        PlainPermissionManager plainPermissionManager = new PlainPermissionManager();
+
+        String samefilePath = file.getAbsolutePath()+"/conf/acl/.";
+        String samefilePath2 = "/" +file.getAbsolutePath()+"/conf/acl";
+        String samefilePath3 = file.getAbsolutePath()+"/conf/acl/../"+file.getAbsolutePath();
+        String samefilePath4 = file.getAbsolutePath()+"/conf/acl///";
+        String samefilePath5 = file.getAbsolutePath()+"/conf/acl/./";
+
+        int size = plainPermissionManager.getDataVersionMap().size();
+
+        plainPermissionManager.load(samefilePath);
+        Assert.assertEquals(size, plainPermissionManager.getDataVersionMap().size());
+
+        plainPermissionManager.load(samefilePath2);
+        Assert.assertEquals(size, plainPermissionManager.getDataVersionMap().size());
+
+        plainPermissionManager.load(samefilePath3);
+        Assert.assertEquals(size, plainPermissionManager.getDataVersionMap().size());
+
+        plainPermissionManager.load(samefilePath4);
+        Assert.assertEquals(size, plainPermissionManager.getDataVersionMap().size());
+
+        plainPermissionManager.load(samefilePath5);
+        Assert.assertEquals(size, plainPermissionManager.getDataVersionMap().size());
+
+    }
+
+    @Test
     public void testWatch() throws IOException, IllegalAccessException, InterruptedException {
         File file = new File("src/test/resources");
         System.setProperty("rocketmq.home.dir", file.getAbsolutePath());
