diff --git a/test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/SQLUtilsTest.java b/test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/SQLUtilsTest.java
index 7f534f5..a006f75 100644
--- a/test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/SQLUtilsTest.java
+++ b/test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/SQLUtilsTest.java
@@ -31,4 +31,17 @@ public class SQLUtilsTest {
         Assert.assertEquals("^A.C$", SQLUtils.makeRegexFromLike("A_C"));
         Assert.assertEquals("A.C", SQLUtils.makeRegexFromLike("%A_C%"));
     }
+    
+    @Test
+    public void fixLineFeedsTest() {
+        Assert.assertEquals(
+            "SELECT LastName -- x\r\n"
+            + "FROM Persons drai where PersonID  = 1\r\n"
+            + "-- AND ResourceId  = 1\n\r"
+            + "ORDER BY PersonID ;",
+            SQLUtils.fixLineFeeds("SELECT LastName -- x\r"
+            + "FROM Persons drai where PersonID  = 1\r\n"
+            + "-- AND ResourceId  = 1\n\r"
+            + "ORDER BY PersonID ;"));
+    }
 }
