diff --git a/test/org.jkiss.dbeaver.ext.greenplum.test/src/org/jkiss/dbeaver/ext/greenplum/model/GreenplumExternalTableTest.java b/test/org.jkiss.dbeaver.ext.greenplum.test/src/org/jkiss/dbeaver/ext/greenplum/model/GreenplumExternalTableTest.java
index 5534a69..57b3b43 100644
--- a/test/org.jkiss.dbeaver.ext.greenplum.test/src/org/jkiss/dbeaver/ext/greenplum/model/GreenplumExternalTableTest.java
+++ b/test/org.jkiss.dbeaver.ext.greenplum.test/src/org/jkiss/dbeaver/ext/greenplum/model/GreenplumExternalTableTest.java
@@ -621,7 +621,7 @@ public class GreenplumExternalTableTest {
     @Test
     public void generateChangeOwnerQuery_whenProvidedAValidOwner_thenShouldGenerateQuerySuccessfully() {
         GreenplumExternalTable table = new GreenplumExternalTable(mockSchema, mockResults);
-        Assert.assertEquals("ALTER EXTERNAL TABLE sampleSchema.sampleTable OWNER TO someOwner",
+        Assert.assertEquals("ALTER EXTERNAL TABLE \"sampleSchema\".\"sampleTable\" OWNER TO someOwner",
                 table.generateChangeOwnerQuery("someOwner"));
     }
 
diff --git a/test/org.jkiss.dbeaver.ext.greenplum.test/src/org/jkiss/dbeaver/ext/greenplum/model/GreenplumTableTest.java b/test/org.jkiss.dbeaver.ext.greenplum.test/src/org/jkiss/dbeaver/ext/greenplum/model/GreenplumTableTest.java
index f61cc73..6a9eeed 100644
--- a/test/org.jkiss.dbeaver.ext.greenplum.test/src/org/jkiss/dbeaver/ext/greenplum/model/GreenplumTableTest.java
+++ b/test/org.jkiss.dbeaver.ext.greenplum.test/src/org/jkiss/dbeaver/ext/greenplum/model/GreenplumTableTest.java
@@ -130,7 +130,7 @@ public class GreenplumTableTest {
 
         table.appendTableModifiers(mockMonitor, ddl);
 
-        Assert.assertEquals("\nDISTRIBUTED BY (Column_Name)", ddl.toString());
+        Assert.assertEquals("\nDISTRIBUTED BY (\"Column_Name\")", ddl.toString());
     }
 
     @Test
@@ -152,7 +152,7 @@ public class GreenplumTableTest {
 
         table.appendTableModifiers(mockMonitor, ddl);
 
-        Assert.assertEquals("\nDISTRIBUTED BY (Column_1, Column_2)", ddl.toString());
+        Assert.assertEquals("\nDISTRIBUTED BY (\"Column_1\", \"Column_2\")", ddl.toString());
     }
 
 
