diff --git a/test/org.jkiss.dbeaver.ext.mysql.test/META-INF/MANIFEST.MF b/test/org.jkiss.dbeaver.ext.mysql.test/META-INF/MANIFEST.MF
new file mode 100644
index 0000000..180bb98
--- /dev/null
+++ b/test/org.jkiss.dbeaver.ext.mysql.test/META-INF/MANIFEST.MF
@@ -0,0 +1,19 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: DBeaver MySQL Tests
+Bundle-SymbolicName: org.jkiss.dbeaver.ext.mysql.test
+Bundle-Version: 1.0.0.qualifier
+Bundle-Release-Date: 20221121
+Bundle-RequiredExecutionEnvironment: JavaSE-11
+Bundle-Vendor: DBeaver Corp
+Bundle-ClassPath: .
+Fragment-Host: org.jkiss.dbeaver.ext.mysql
+Bundle-ActivationPolicy: lazy
+Require-Bundle: org.eclipse.core.runtime,
+ org.eclipse.core.resources,
+ org.junit,
+ org.mockito.mockito-core,
+ org.jkiss.dbeaver.headless,
+ org.jkiss.dbeaver.model,
+ org.jkiss.dbeaver.registry,
+ org.jkiss.dbeaver.ext.mysql
diff --git a/test/org.jkiss.dbeaver.ext.mysql.test/build.properties b/test/org.jkiss.dbeaver.ext.mysql.test/build.properties
new file mode 100644
index 0000000..2aee438
--- /dev/null
+++ b/test/org.jkiss.dbeaver.ext.mysql.test/build.properties
@@ -0,0 +1,4 @@
+source.. = src/
+output.. = target/classes/
+bin.includes = .,\
+               META-INF/
diff --git a/test/org.jkiss.dbeaver.ext.mysql.test/pom.xml b/test/org.jkiss.dbeaver.ext.mysql.test/pom.xml
new file mode 100644
index 0000000..dc7e5ca
--- /dev/null
+++ b/test/org.jkiss.dbeaver.ext.mysql.test/pom.xml
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.jkiss.dbeaver</groupId>
+        <artifactId>tests</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+        <relativePath>../</relativePath>
+    </parent>
+    <artifactId>org.jkiss.dbeaver.ext.mysql.test</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <packaging>eclipse-test-plugin</packaging>
+
+</project>
diff --git a/test/org.jkiss.dbeaver.ext.mysql.test/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialectTest.java b/test/org.jkiss.dbeaver.ext.mysql.test/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialectTest.java
new file mode 100644
index 0000000..7299117
--- /dev/null
+++ b/test/org.jkiss.dbeaver.ext.mysql.test/src/org/jkiss/dbeaver/ext/mysql/model/MySQLDialectTest.java
@@ -0,0 +1,34 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.mysql.model;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MySQLDialectTest {
+    @Test
+    public void quoteStringTest() {
+        final MySQLDialect dialect = new MySQLDialect();
+
+        Assert.assertEquals("`a ' b`", dialect.getQuotedIdentifier("a ' b", false, true));
+        Assert.assertEquals("`a `` b`", dialect.getQuotedIdentifier("a ` b", false, true));
+        Assert.assertEquals("`a b`", dialect.getQuotedIdentifier("a b", false, true));
+    }
+}
