diff --git a/test/framework/build.gradle b/test/framework/build.gradle
index 81df9b9..826fdb7 100644
--- a/test/framework/build.gradle
+++ b/test/framework/build.gradle
@@ -93,5 +93,5 @@ tasks.register("verifyVersions") {
 
 tasks.named('splitPackagesAudit').configure {
   // for now we always run tests with the classpath, so we are ok with split packages for tests
-  onlyIf { false }
+  enabled = false
 }
