diff --git a/qa/apm/src/test/java/org/elasticsearch/tracing/apm/ApmIT.java b/qa/apm/src/test/java/org/elasticsearch/tracing/apm/ApmIT.java
index a6bf3ac..41421a4 100644
--- a/qa/apm/src/test/java/org/elasticsearch/tracing/apm/ApmIT.java
+++ b/qa/apm/src/test/java/org/elasticsearch/tracing/apm/ApmIT.java
@@ -46,7 +46,7 @@ public class ApmIT extends ESRestTestCase {
      * </ul>
      * <p>This proves that the hierarchy of spans is being correctly captured.
      */
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/89340")
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/90308")
     public void testCapturesTracesForHttpTraffic() throws Exception {
         checkTracesDataStream();
 
