diff --git a/sslcontext-kickstart-for-pem/src/test/java/nl/altindag/ssl/util/PemUtilsShould.java b/sslcontext-kickstart-for-pem/src/test/java/nl/altindag/ssl/util/PemUtilsShould.java
index 703da74..a675017 100644
--- a/sslcontext-kickstart-for-pem/src/test/java/nl/altindag/ssl/util/PemUtilsShould.java
+++ b/sslcontext-kickstart-for-pem/src/test/java/nl/altindag/ssl/util/PemUtilsShould.java
@@ -270,6 +270,16 @@ class PemUtilsShould {
     }
 
     @Test
+    void loadEncryptedPrivateKeyShouldFailWhenMissingPassword() {
+        assertThatThrownBy(() -> PemUtils.loadIdentityMaterial(
+                PEM_LOCATION + "splitted-encrypted-identity-containing-certificate.pem",
+                PEM_LOCATION + "splitted-encrypted-identity-containing-private-key.pem",
+                null))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("A password is mandatory with encrypted private key informations");
+    }
+
+    @Test
     void loadRsaUnencryptedIdentityMaterialFromClassPath() {
         X509ExtendedKeyManager keyManager = PemUtils.loadIdentityMaterial(PEM_LOCATION + "rsa-unencrypted-identity.pem");
 
