diff --git a/brut.apktool/apktool-lib/src/test/java/brut/androlib/util/UnknownDirectoryTraversalTest.java b/brut.apktool/apktool-lib/src/test/java/brut/androlib/util/UnknownDirectoryTraversalTest.java
index 5c61877..ad22e9d 100644
--- a/brut.apktool/apktool-lib/src/test/java/brut/androlib/util/UnknownDirectoryTraversalTest.java
+++ b/brut.apktool/apktool-lib/src/test/java/brut/androlib/util/UnknownDirectoryTraversalTest.java
@@ -26,6 +26,7 @@ import brut.directory.ExtFile;
 import brut.util.BrutIO;
 import brut.util.OS;
 import brut.util.OSDetection;
+import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
 
@@ -43,6 +44,11 @@ public class UnknownDirectoryTraversalTest extends BaseTest {
         TestUtils.copyResourceDir(UnknownDirectoryTraversalTest.class, "util/traversal", sTmpDir);
     }
 
+    @AfterClass
+    public static void afterClass() throws BrutException {
+        OS.rmdir(sTmpDir);
+    }
+
     @Test
     public void validFileTest() throws IOException, BrutException {
         String validFilename = BrutIO.sanitizeUnknownFile(sTmpDir, "file");
