diff --git a/singleton/src/test/java/com/iluwatar/singleton/BillPughImplementationTest.java b/singleton/src/test/java/com/iluwatar/singleton/BillPughImplementationTest.java
new file mode 100644
index 0000000..90a3424
--- /dev/null
+++ b/singleton/src/test/java/com/iluwatar/singleton/BillPughImplementationTest.java
@@ -0,0 +1,16 @@
+package com.iluwatar.singleton;
+
+/**
+ * Date: 06/18/23 - 16:29 PM.
+ *
+ * @author Owen Leung
+ */
+public class BillPughImplementationTest
+        extends SingletonTest<BillPughImplementation>{
+    /**
+     * Create a new singleton test instance using the given 'getInstance' method.
+     */
+    public BillPughImplementationTest() {
+        super(BillPughImplementation::getInstance);
+    }
+}
