diff --git a/javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/Issue2374Test.java b/javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/Issue2374Test.java
index 469d930..d760c33 100644
--- a/javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/Issue2374Test.java
+++ b/javaparser-core-testing/src/test/java/com/github/javaparser/printer/lexicalpreservation/Issue2374Test.java
@@ -29,6 +29,7 @@ import org.junit.jupiter.api.Test;
 import java.util.Optional;
 
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static com.github.javaparser.utils.TestUtils.assertEqualsStringIgnoringEol;
 
 public class Issue2374Test extends AbstractLexicalPreservingTest {
     
@@ -42,6 +43,14 @@ public class Issue2374Test extends AbstractLexicalPreservingTest {
                 "    }\n" + 
                 "}"
                 );
+        String expected =
+        		"public class Bar {\n"
+        		+ "    public void foo() {\n"
+        		+ "        System.out.print(\"Hello\");\n"
+        		+ "        //Example comment\n"
+        		+ "        System.out.println(\"World!\");\n"
+        		+ "    }\n"
+        		+ "}";
         // contruct a statement with a comment
         Statement stmt = StaticJavaParser.parseStatement("System.out.println(\"World!\");");
         stmt.setLineComment(lineComment);
@@ -51,6 +60,6 @@ public class Issue2374Test extends AbstractLexicalPreservingTest {
         // print the result from LexicalPreservingPrinter
         String result = LexicalPreservingPrinter.print(cu);
         // verify that the LexicalPreservingPrinter don't forget the comment
-        assertTrue(result.contains(lineComment));
+        assertEqualsStringIgnoringEol(expected, result);
     }
 }
