diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/AbstractMigrationTest.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/AbstractMigrationTest.java
index 9bd3bc7..d2c6fe3 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/AbstractMigrationTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/AbstractMigrationTest.java
@@ -59,6 +59,7 @@ import org.keycloak.representations.idm.ProtocolMapperRepresentation;
 import org.keycloak.representations.idm.RealmRepresentation;
 import org.keycloak.representations.idm.RequiredActionProviderRepresentation;
 import org.keycloak.representations.idm.RoleRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
 import org.keycloak.representations.idm.authorization.DecisionStrategy;
 import org.keycloak.representations.idm.authorization.ResourceRepresentation;
 import org.keycloak.representations.idm.authorization.ResourceServerRepresentation;
@@ -84,12 +85,14 @@ import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+import static org.hamcrest.CoreMatchers.not;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasEntry;
 import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.notNullValue;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
@@ -320,11 +323,17 @@ public abstract class AbstractMigrationTest extends AbstractKeycloakTest {
         testPostLogoutRedirectUrisSet(migrationRealm);
     }
 
-   protected void testMigrationTo20_0_0() {
+    protected void testMigrationTo20_0_0() {
         testViewGroups(masterRealm);
         testViewGroups(migrationRealm);
     }
 
+   protected void testMigrationTo21_0_2() {
+       testTermsAndConditionsMigrated(masterRealm);
+       testTermsAndConditionsMigrated(migrationRealm);
+       testTermsAndConditionsMigrated(migrationRealm2);
+   }
+
 
     protected void testDeleteAccount(RealmResource realm) {
         ClientRepresentation accountClient = realm.clients().findByClientId(ACCOUNT_MANAGEMENT_CLIENT_ID).get(0);
@@ -505,6 +514,32 @@ public abstract class AbstractMigrationTest extends AbstractKeycloakTest {
         assertNotNull(viewAppRole);
     }
 
+    protected void testTermsAndConditionsMigrated(RealmResource realmResource) {
+        final String legacyTermsAndConditionsAlias = "terms_and_conditions";
+        // Test realm RequiredAction migrated
+        RealmRepresentation realm = realmResource.toRepresentation();
+        List<RequiredActionProviderRepresentation> requiredActions = realm.getRequiredActions();
+
+        if (requiredActions != null && !requiredActions.isEmpty()) {
+            assertThat(requiredActions.stream()
+                    .map(RequiredActionProviderRepresentation::getAlias)
+                    .collect(Collectors.toList()), not(hasItem(legacyTermsAndConditionsAlias)));
+            assertThat(requiredActions.stream()
+                    .map(RequiredActionProviderRepresentation::getProviderId)
+                    .collect(Collectors.toList()), not(hasItem(legacyTermsAndConditionsAlias)));
+        }
+
+        List<UserRepresentation> users = realmResource.users().list(null, null);
+
+        if (users != null && !users.isEmpty()) {
+            // Test users required actions migrated
+            assertThat(users.stream()
+                            .flatMap(user -> user.getRequiredActions().stream())
+                            .collect(Collectors.toList()),
+                    not(hasItem(legacyTermsAndConditionsAlias)));
+        }
+    }
+
     protected void testRoleManageAccountLinks(RealmResource... realms) {
         log.info("testing role manage account links");
         for (RealmResource realm : realms) {
@@ -972,6 +1007,10 @@ public abstract class AbstractMigrationTest extends AbstractKeycloakTest {
         testMigrationTo20_0_0();
     }
 
+    protected void testMigrationTo21_x() {
+        testMigrationTo21_0_2();
+    }
+
     protected void testMigrationTo7_x(boolean supportedAuthzServices) {
         if (supportedAuthzServices) {
             testDecisionStrategySetOnResourceServer();
diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport198MigrationTest.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport198MigrationTest.java
index 9f775d0..a30743b 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport198MigrationTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport198MigrationTest.java
@@ -75,6 +75,7 @@ public class JsonFileImport198MigrationTest extends AbstractJsonFileImportMigrat
         testMigrationTo12_x(false);
         testMigrationTo18_x();
         testMigrationTo20_x();
+        testMigrationTo21_x();
     }
 
     @Override
diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport255MigrationTest.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport255MigrationTest.java
index 8b77488..3f803b8 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport255MigrationTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport255MigrationTest.java
@@ -69,6 +69,7 @@ public class JsonFileImport255MigrationTest extends AbstractJsonFileImportMigrat
         testMigrationTo12_x(false);
         testMigrationTo18_x();
         testMigrationTo20_x();
+        testMigrationTo21_x();
     }
 
 }
diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport343MigrationTest.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport343MigrationTest.java
index 40728b3..59063f5 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport343MigrationTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport343MigrationTest.java
@@ -64,6 +64,7 @@ public class JsonFileImport343MigrationTest extends AbstractJsonFileImportMigrat
         testMigrationTo12_x(true);
         testMigrationTo18_x();
         testMigrationTo20_x();
+        testMigrationTo21_x();
     }
 
 }
diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport483MigrationTest.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport483MigrationTest.java
index a007799..525b8b1 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport483MigrationTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport483MigrationTest.java
@@ -58,6 +58,7 @@ public class JsonFileImport483MigrationTest extends AbstractJsonFileImportMigrat
         testMigrationTo12_x(true);
         testMigrationTo18_x();
         testMigrationTo20_x();
+        testMigrationTo21_x();
     }
 
 }
diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport903MigrationTest.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport903MigrationTest.java
index 322e16b..80e7862 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport903MigrationTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport903MigrationTest.java
@@ -51,6 +51,7 @@ public class JsonFileImport903MigrationTest extends AbstractJsonFileImportMigrat
         testMigrationTo12_x(true);
         testMigrationTo18_x();
         testMigrationTo20_x();
+        testMigrationTo21_x();
     }
 
 }
diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/MigrationTest.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/MigrationTest.java
index eeb5e44..a0a186b 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/MigrationTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/MigrationTest.java
@@ -69,5 +69,6 @@ public class MigrationTest extends AbstractMigrationTest {
         testExtremelyLongClientAttribute(migrationRealm);
 
         testMigrationTo20_x();
+        testMigrationTo21_x();
     }
 }
