diff --git a/quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildAndStartDistTest.java b/quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildAndStartDistTest.java
index cfe612d..5682f21 100644
--- a/quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildAndStartDistTest.java
+++ b/quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildAndStartDistTest.java
@@ -17,15 +17,19 @@
 
 package org.keycloak.it.cli.dist;
 
+import io.quarkus.test.junit.main.Launch;
+import io.quarkus.test.junit.main.LaunchResult;
 import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestMethodOrder;
 import org.keycloak.it.junit5.extension.CLIResult;
 import org.keycloak.it.junit5.extension.DistributionTest;
 import org.keycloak.it.junit5.extension.RawDistOnly;
+import org.keycloak.it.junit5.extension.WithEnvVars;
 import org.keycloak.it.utils.KeycloakDistribution;
 import org.keycloak.it.utils.RawKeycloakDistribution;
 
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG;
 
 @DistributionTest
@@ -63,4 +67,30 @@ public class BuildAndStartDistTest {
         cliResult.assertBuild();
         cliResult.assertStarted();
     }
+
+    @Test
+    @WithEnvVars({"KEYCLOAK_ADMIN", "admin123", "KEYCLOAK_ADMIN_PASSWORD", "admin123"})
+    @Launch({"start-dev"})
+    void testCreateAdmin(KeycloakDistribution dist, LaunchResult result) {
+        assertAdminCreation(dist, result, "admin123", "admin123", "admin123");
+    }
+
+    @Test
+    @WithEnvVars({"KEYCLOAK_ADMIN", "admin123", "KEYCLOAK_ADMIN_PASSWORD", "admin123"})
+    @Launch({"start-dev"})
+    void testCreateDifferentAdmin(KeycloakDistribution dist, LaunchResult result) {
+        assertAdminCreation(dist, result, "admin123", "new-admin", "new-admin");
+    }
+
+    private void assertAdminCreation(KeycloakDistribution dist, LaunchResult result, String initialUsername, String nextUsername, String password) {
+        assertTrue(result.getOutput().contains("Added user '" + initialUsername + "' to realm 'master'"),
+                () -> "The Output:\n" + result.getOutput() + "doesn't contains the expected string.");
+
+        dist.setEnvVar("KEYCLOAK_ADMIN", nextUsername);
+        dist.setEnvVar("KEYCLOAK_ADMIN_PASSWORD", password);
+        CLIResult cliResult = dist.run("start-dev", "--log-level=debug");
+
+        cliResult.assertMessage("Skipping create admin user. Admin already exists in realm 'master'.");
+        cliResult.assertStartedDevMode();
+    }
 }
