diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/util/AccountHelper.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/util/AccountHelper.java
index 934de3c..4125055 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/util/AccountHelper.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/util/AccountHelper.java
@@ -31,7 +31,7 @@ import java.util.Optional;
 public class AccountHelper {
 
     private static UserResource getUserResource(RealmResource realm, String username) {
-        Optional<UserRepresentation> userResult = realm.users().search(username).stream().findFirst();
+        Optional<UserRepresentation> userResult = realm.users().search(username, true).stream().findFirst();
         if (userResult.isEmpty()) {
             throw new RuntimeException("User with username " + username + " not found");
         }
