diff --git a/testsuite/model/src/test/java/org/keycloak/testsuite/model/session/UserSessionProviderModelTest.java b/testsuite/model/src/test/java/org/keycloak/testsuite/model/session/UserSessionProviderModelTest.java
index ff0c2f2..b19e6b6 100644
--- a/testsuite/model/src/test/java/org/keycloak/testsuite/model/session/UserSessionProviderModelTest.java
+++ b/testsuite/model/src/test/java/org/keycloak/testsuite/model/session/UserSessionProviderModelTest.java
@@ -263,6 +263,7 @@ public class UserSessionProviderModelTest extends KeycloakModelTest {
             RemoteCache<String, HotRodUserSessionEntity> remoteCache = provider.getRemoteCache(ModelEntityUtil.getModelName(UserSessionModel.class));
             HotRodUserSessionEntity userSessionEntity = new HotRodUserSessionEntity();
             userSessionEntity.id = UUID.randomUUID().toString();
+            userSessionEntity.realmId = realmId;
             remoteCache.put(userSessionEntity.id, userSessionEntity);
         }));
 
