diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java
index 2402921..00574a2 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java
@@ -383,7 +383,6 @@ public class JavascriptAdapterTest extends AbstractJavascriptTest {
                 .update()
         ) {
                 setImplicitFlowForClient();
-                testExecutor.init(defaultArguments().implicitFlow());
 
                 testExecutor.logInAndInit(defaultArguments().implicitFlow(), testUser, this::assertInitAuth);
                 assertThat(driver.getPageSource(), not(containsString("Access token expired")));
