diff --git a/operator/src/test/java/org/keycloak/operator/testsuite/integration/ClusteringTest.java b/operator/src/test/java/org/keycloak/operator/testsuite/integration/ClusteringTest.java
index 8167df1..38e85e7 100644
--- a/operator/src/test/java/org/keycloak/operator/testsuite/integration/ClusteringTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/integration/ClusteringTest.java
@@ -92,11 +92,6 @@ public class ClusteringTest extends BaseOperatorTest {
                 .ignoreExceptions()
                 .untilAsserted(() -> assertThat(kcPodsSelector.list().getItems().size()).isEqualTo(3));
 
-        Awaitility.await()
-                .atMost(Duration.ofSeconds(60))
-                .ignoreExceptions()
-                .untilAsserted(() -> assertThat(crSelector.scale().getStatus().getReplicas()).isEqualTo(3));
-
         // when scale it down to 2
         crSelector.scale(2);
         assertThat(crSelector.scale().getSpec().getReplicas()).isEqualTo(2);
