diff --git a/federation/ldap/src/test/java/org/keycloak/storage/ldap/idm/model/EscapeTest.java b/federation/ldap/src/test/java/org/keycloak/storage/ldap/idm/model/EscapeTest.java
index 64cb641..8950edb 100644
--- a/federation/ldap/src/test/java/org/keycloak/storage/ldap/idm/model/EscapeTest.java
+++ b/federation/ldap/src/test/java/org/keycloak/storage/ldap/idm/model/EscapeTest.java
@@ -27,16 +27,16 @@ import org.keycloak.storage.ldap.idm.query.EscapeStrategy;
 public class EscapeTest {
 
     @Test
-    public void testNoAsciiOnlyEscaping() throws Exception {
+    public void testEscapingExceptAsterisk() {
         String text = "Véronique* Martin(john)second\\fff//eee\u0000";
-        Assert.assertEquals(EscapeStrategy.NON_ASCII_CHARS_ONLY.escape(text), "V\\c3\\a9ronique* Martin(john)second\\fff//eee\u0000");
+        Assert.assertEquals(EscapeStrategy.DEFAULT_EXCEPT_ASTERISK.escape(text), "V\\c3\\a9ronique* Martin\\28john\\29second\\5cfff//eee\\00");
 
         text = "Hi This is a test #çà";
-        Assert.assertEquals(EscapeStrategy.DEFAULT.escape(text), "Hi This is a test #\\c3\\a7\\c3\\a0");
+        Assert.assertEquals(EscapeStrategy.DEFAULT_EXCEPT_ASTERISK.escape(text), "Hi This is a test #\\c3\\a7\\c3\\a0");
     }
 
     @Test
-    public void testEscaping() throws Exception {
+    public void testEscaping() {
         String text = "Véronique* Martin(john)second\\fff//eee\u0000";
         Assert.assertEquals(EscapeStrategy.DEFAULT.escape(text), "V\\c3\\a9ronique\\2a Martin\\28john\\29second\\5cfff//eee\\00");
 
