diff --git a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcCustomOidcBrokerTest.java b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcCustomOidcBrokerTest.java
index 01d0665..4503cb0 100644
--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcCustomOidcBrokerTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcCustomOidcBrokerTest.java
@@ -20,11 +20,12 @@ import static org.keycloak.testsuite.broker.BrokerTestConstants.IDP_OIDC_ALIAS;
 import static org.keycloak.testsuite.broker.BrokerTestTools.createIdentityProvider;
 import static org.keycloak.testsuite.broker.BrokerTestTools.getConsumerRoot;
 
-import org.junit.Assert;
 import org.junit.Test;
 import org.keycloak.models.IdentityProviderSyncMode;
 import org.keycloak.representations.idm.IdentityProviderRepresentation;
 import org.keycloak.testsuite.broker.oidc.TestKeycloakOidcIdentityProviderFactory;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
 
 /**
  * Test methods for testing a custom OIDC broker
@@ -51,6 +52,6 @@ public class KcCustomOidcBrokerTest extends AbstractInitializedBaseBrokerTest {
     @Test
     public void testCustomDisplayIcon() {
         driver.navigate().to(getAccountUrl(getConsumerRoot(), bc.consumerRealmName()));
-        Assert.assertTrue(driver.getPageSource().contains("my-custom-idp-icon"));
+        assertThat(driver.getPageSource(), containsString("my-custom-idp-icon"));
     }
 }
