diff --git a/src/test/java/org/mockitousage/basicapi/MocksSerializationForAnnotationTest.java b/src/test/java/org/mockitousage/basicapi/MocksSerializationForAnnotationTest.java
index 736de4e..68e54f9 100644
--- a/src/test/java/org/mockitousage/basicapi/MocksSerializationForAnnotationTest.java
+++ b/src/test/java/org/mockitousage/basicapi/MocksSerializationForAnnotationTest.java
@@ -13,6 +13,7 @@ import java.io.Serializable;
 import java.util.Collections;
 import java.util.List;
 import java.util.Observable;
+import java.util.function.Supplier;
 
 import org.assertj.core.api.Assertions;
 import org.junit.Test;
@@ -53,6 +54,9 @@ public class MocksSerializationForAnnotationTest extends TestBase implements Ser
             serializable = true)
     IMethods imethodsWithExtraInterfacesMock;
 
+    @Mock(serializable = true)
+    Supplier<Object> parameterizedSupplier;
+
     @Test
     public void should_allow_throws_exception_to_be_serializable() throws Exception {
         // given
@@ -69,6 +73,11 @@ public class MocksSerializationForAnnotationTest extends TestBase implements Ser
     }
 
     @Test
+    public void should_allow_mock_of_parameterized_type_to_be_serializable() throws Exception {
+        serializeAndBack(parameterizedSupplier);
+    }
+
+    @Test
     public void should_allow_mock_and_boolean_value_to_serializable() throws Exception {
         // given
         when(imethodsMock.booleanReturningMethod()).thenReturn(true);
