diff --git a/codec-http/src/test/java/io/netty/handler/codec/http/multipart/MixedTest.java b/codec-http/src/test/java/io/netty/handler/codec/http/multipart/MixedTest.java
index b10e144..2e6e6a8 100644
--- a/codec-http/src/test/java/io/netty/handler/codec/http/multipart/MixedTest.java
+++ b/codec-http/src/test/java/io/netty/handler/codec/http/multipart/MixedTest.java
@@ -20,8 +20,12 @@ import io.netty.util.CharsetUtil;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+import java.io.File;
 import java.io.IOException;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
 public class MixedTest {
     @Test
     public void mixedAttributeRefCnt() throws IOException {
@@ -54,4 +58,19 @@ public class MixedTest {
 
         upload.release(2);
     }
+
+    @Test
+    public void testSpecificCustomBaseDir() throws IOException {
+        File baseDir = new File("target/MixedTest/testSpecificCustomBaseDir");
+        baseDir.mkdirs(); // we don't need to clean it since it is in volatile files anyway
+        MixedFileUpload upload = new MixedFileUpload("foo", "foo", "foo", "UTF-8", CharsetUtil.UTF_8, 1000, 100,
+                                                     baseDir.getAbsolutePath(), true);
+
+        upload.addContent(Unpooled.wrappedBuffer(new byte[1000]), true);
+
+        assertTrue(upload.getFile().getAbsolutePath().startsWith(baseDir.getAbsolutePath()));
+        assertTrue(upload.getFile().exists());
+        assertEquals(1000, upload.getFile().length());
+        upload.delete();
+    }
 }
