diff --git a/redisson/src/test/java/org/redisson/RedissonLockTest.java b/redisson/src/test/java/org/redisson/RedissonLockTest.java
index b412d9d..414f46f 100644
--- a/redisson/src/test/java/org/redisson/RedissonLockTest.java
+++ b/redisson/src/test/java/org/redisson/RedissonLockTest.java
@@ -112,6 +112,29 @@ public class RedissonLockTest extends BaseConcurrentTest {
     }
 
     @Test
+    public void testLockIsNotRenewedAfterInterruptedTryLock() throws InterruptedException {
+        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        RLock lock = redisson.getLock("myLock");
+        assertThat(lock.isLocked()).isFalse();
+
+        Thread thread = new Thread(() -> {
+            countDownLatch.countDown();
+            if (!lock.tryLock()) {
+                return;
+            }
+            lock.unlock();
+        });
+        thread.start();
+        countDownLatch.await();
+        // let the tcp request be sent out
+        TimeUnit.MILLISECONDS.sleep(5);
+        thread.interrupt();
+        TimeUnit.SECONDS.sleep(45);
+
+        assertThat(lock.isLocked()).isFalse();
+    }
+
+    @Test
     public void testRedisFailed() {
         Assertions.assertThrows(WriteRedisConnectionException.class, () -> {
             RedisRunner.RedisProcess master = new RedisRunner()
