diff --git a/discovery/seata-discovery-core/src/test/java/io/seata/discovery/loadbalance/LoadBalanceTest.java b/discovery/seata-discovery-core/src/test/java/io/seata/discovery/loadbalance/LoadBalanceTest.java
index 41a88e3..0b113c1 100644
--- a/discovery/seata-discovery-core/src/test/java/io/seata/discovery/loadbalance/LoadBalanceTest.java
+++ b/discovery/seata-discovery-core/src/test/java/io/seata/discovery/loadbalance/LoadBalanceTest.java
@@ -68,6 +68,26 @@ public class LoadBalanceTest {
     }
 
     /**
+     * Test xid load load balance select.
+     *
+     * @param addresses the addresses
+     */
+    @ParameterizedTest
+    @MethodSource("addressProvider")
+    public void testXIDLoadBalance_select(List<InetSocketAddress> addresses) throws Exception {
+        XIDLoadBalance loadBalance = new XIDLoadBalance();
+        // ipv4
+        InetSocketAddress inetSocketAddress = loadBalance.select(addresses, "127.0.0.1:8092:123456");
+        Assertions.assertNotNull(inetSocketAddress);
+        // ipv6
+        inetSocketAddress = loadBalance.select(addresses, "2000:0000:0000:0000:0001:2345:6789:abcd:8092:123456");
+        Assertions.assertNotNull(inetSocketAddress);
+        // test not found tc channel
+        inetSocketAddress = loadBalance.select(addresses, "127.0.0.1:8199:123456");
+        Assertions.assertNotEquals(inetSocketAddress.getPort(),8199);
+    }
+
+    /**
      * Test consistent hash load load balance select.
      *
      * @param addresses the addresses
@@ -156,7 +176,8 @@ public class LoadBalanceTest {
                         new InetSocketAddress("127.0.0.1", 8092),
                         new InetSocketAddress("127.0.0.1", 8093),
                         new InetSocketAddress("127.0.0.1", 8094),
-                        new InetSocketAddress("127.0.0.1", 8095))
+                        new InetSocketAddress("127.0.0.1", 8095),
+                        new InetSocketAddress("2000:0000:0000:0000:0001:2345:6789:abcd", 8092))
         );
     }
 }
