diff --git a/server/src/test/java/io/seata/server/session/GlobalSessionTest.java b/server/src/test/java/io/seata/server/session/GlobalSessionTest.java
index 25eb086..b631d41 100644
--- a/server/src/test/java/io/seata/server/session/GlobalSessionTest.java
+++ b/server/src/test/java/io/seata/server/session/GlobalSessionTest.java
@@ -21,7 +21,6 @@ import java.util.stream.Stream;
 import io.seata.core.model.BranchStatus;
 import io.seata.core.model.BranchType;
 import io.seata.core.model.GlobalStatus;
-import io.seata.server.storage.file.session.FileSessionManager;
 import io.seata.server.store.StoreConfig;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.Assertions;
@@ -46,11 +45,7 @@ public class GlobalSessionTest {
 
 
     @BeforeAll
-    public static void setUp(ApplicationContext context){
-
-    }
-    @BeforeAll
-    public static void init(){
+    public static void init(ApplicationContext context){
         SessionHolder.init(StoreConfig.SessionMode.FILE);
     }
     @AfterAll
diff --git a/server/src/test/java/io/seata/server/store/file/FileTransactionStoreManagerTest.java b/server/src/test/java/io/seata/server/store/file/FileTransactionStoreManagerTest.java
index 6315888..f311482 100644
--- a/server/src/test/java/io/seata/server/store/file/FileTransactionStoreManagerTest.java
+++ b/server/src/test/java/io/seata/server/store/file/FileTransactionStoreManagerTest.java
@@ -35,6 +35,7 @@ import org.assertj.core.util.Files;
 import org.junit.jupiter.api.*;
 import org.mockito.Mockito;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContext;
 
 /**
  * @author ggndnn
@@ -43,7 +44,7 @@ import org.springframework.boot.test.context.SpringBootTest;
 public class FileTransactionStoreManagerTest {
 
     @BeforeAll
-    public static void init(){
+    public static void init(ApplicationContext context){
         SessionHolder.init(StoreConfig.SessionMode.FILE);
     }
     @AfterAll
