diff --git a/terminal-emulator/src/test/java/com/termux/terminal/TerminalTestCase.java b/terminal-emulator/src/test/java/com/termux/terminal/TerminalTestCase.java
index eb1da8c..db93885 100644
--- a/terminal-emulator/src/test/java/com/termux/terminal/TerminalTestCase.java
+++ b/terminal-emulator/src/test/java/com/termux/terminal/TerminalTestCase.java
@@ -108,7 +108,7 @@ public abstract class TerminalTestCase extends TestCase {
 
 	protected TerminalTestCase withTerminalSized(int columns, int rows) {
 	    // The tests aren't currently using the client, so a null client will suffice, a dummy client should be implemented if needed
-		mTerminal = new TerminalEmulator(mOutput, columns, rows, rows * 2, null);
+		mTerminal = new TerminalEmulator(mOutput, false, columns, rows, rows * 2, null);
 		return this;
 	}
 
