diff --git a/src/test/java/com/thealgorithms/strings/WordLadderTest.java b/src/test/java/com/thealgorithms/strings/WordLadderTest.java
new file mode 100644
index 0000000..4e1e1db
--- /dev/null
+++ b/src/test/java/com/thealgorithms/strings/WordLadderTest.java
@@ -0,0 +1,17 @@
+package com.thealgorithms.strings;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+import java.util.*;
+
+public class WordLadderTest {
+
+    @Test
+    public void testWordLadder() {
+        String words1[] = { "hot", "dot", "dog", "lot", "log", "cog" };
+        assertEquals(5, WordLadder.ladderLength("hit", "cog", Arrays.asList(words1)));
+        String words2[] = { "hot", "dot", "dog", "lot", "log" };
+        assertEquals(0, WordLadder.ladderLength("hit", "cog", Arrays.asList(words2)));
+     }
+}
