diff --git a/src/test/java/com/thealgorithms/strings/CheckVowelsTest.java b/src/test/java/com/thealgorithms/strings/CheckVowelsTest.java
new file mode 100644
index 0000000..7491892
--- /dev/null
+++ b/src/test/java/com/thealgorithms/strings/CheckVowelsTest.java
@@ -0,0 +1,16 @@
+package com.thealgorithms.strings;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+public class CheckVowelsTest {
+
+    @Test
+    public void isVowel() {
+        assertTrue(CheckVowels.hasVowels("foo"));
+        assertTrue(CheckVowels.hasVowels("bar"));
+        assertFalse(CheckVowels.hasVowels("why"));
+        assertFalse(CheckVowels.hasVowels("myths"));
+    }
+}
