diff --git a/src/test/java/com/thealgorithms/strings/RotationTest.java b/src/test/java/com/thealgorithms/strings/RotationTest.java
new file mode 100644
index 0000000..781a68d
--- /dev/null
+++ b/src/test/java/com/thealgorithms/strings/RotationTest.java
@@ -0,0 +1,15 @@
+package com.thealgorithms.strings;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+public class RotationTest {
+
+    @Test
+    public void testRotation() {
+        assertEquals("eksge", Rotation.rotation("geeks", 2));
+        assertEquals("anasban", Rotation.rotation("bananas", 3));
+        assertEquals("abracadabra", Rotation.rotation("abracadabra", 0));
+     }
+}
